---
title: "Gene Set Enrichment Analysis- Maternal Liver Transcriptome Data"
author: "Kaylee Helfrich"
date: "7/22/2019 - 7/29/2019"
output: 
  pdf_document: default
  html_document: default
data: DEG_Results_MDvAlc_MatLiv_LFCShrink_VSTParam_RemovedNA_ExcelProof.txt
---
Built with R `r getRversion()`

```{r RMarkdown setup, include = FALSE, message = FALSE}
require(knitr)
knitr::opts_chunk$set(echo = TRUE, tidy.opts=list(width.cutoff=60), tidy=TRUE)
knitr::opts_knit$set(root.dir = "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/GSEA")
options(tinytex.verbose = TRUE)
```

# Gene Set Enrichment Analysis
This document details my code and output for the gene set enrichment analysis (GSEA) on maternal liver transcriptome data. Litter #'s: A1,A2,A3,A4,A5,A6,A7,A8,B1,B2,B4,B5,B6,B7,B8,B9 (where A is MD and B is Alc). 3g/kg alcohol as oral gavage (100% MD, split dose) from E8.5-17.5. Generated by KH, CK, NS Apr-June 2018.

```{r include = FALSE}
rm(list=ls())
setwd("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/GSEA")
library("ggplot2")
library("BiocParallel")
library("clusterProfiler")
library("org.Mm.eg.db")
library("DOSE")
library("pathview")
library("AnnotationHub")
library("ensembldb")
library("tidyverse")
library("annotables")
library("dplyr")
```

The gene set enrichment package used here requires entrez IDs, so we import the file, clean it, annotate it with the correct IDs, and then extract out the ones with the Entrez IDs along with their fold changes.
```{r Load file and add Entrez IDs and sort by fold change, message = FALSE, warning = FALSE}
file <- read.table("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/GSEA/DEG_Results_MDvAlc_MatLiv_LFCShrink_VSTParam_RemovedNA_ExcelProof.txt", header = TRUE, sep = "")
file <- file[,-1] #remove first column which has meaningless numbers
idx <- grcm38$symbol %in% file$GeneID #Return the IDs for the gene symbols in the DE results
ids <- grcm38[idx, ] #new dataframe with the IDs from the DE results that pulls all info from grcm38
#remove any duplicated symbols since gene names can map to more than one Ensembl ID
non_duplicates <- which(duplicated(ids$symbol) == FALSE) 
ids <- ids[non_duplicates, ] #add only the nonduplicated symbols to the file
#add the gene IDs and other information to the DEG results
file_ids <- inner_join(file, ids, by=c("GeneID"="symbol")) 
file_ids <- file_ids[,c(1,23:30,2:22)] #switch the order of the columns
file_entrez <- filter(file_ids, entrez != "NA") #remove any NA values
file_entrez <- file_entrez[which(duplicated(file_entrez$entrez) == F), ] #remove any Entrez duplicates
foldchanges <- file_entrez$log2FoldChange #Extract foldchanges
names(foldchanges) <- as.character(file_entrez$entrez) #Name each fold change with its Entrez ID
foldchanges <- sort(foldchanges, decreasing = TRUE) #sort gene list into descending list for the GSEA which requires a sorted list
```

## Summary of genes used for analysis
I imported a list of **14,788** genes total. Of that list, **14,137** were annotated with information from the grcm38 build. Of those genes, **13,918** were non-duplicates (and the duplicates were removed). After removing NA values from the annotated Entrez ID list, **13,207** genes remained. This list of **13,207** genes were used for the analysis below.

## Gene Set Enrichment Analysis with ClusterProfiler
Gene Set Enrichment Analysis (GSEA) is a form of analysis that uses log2fold change data from all of the genes measured and then determines whether gene sets for certain biological pathways are significantly enriched among the large positive or negative fold changes. This type of analysis gets around the problem of not seeing large fold changes. This analysis works well when there are smaller but coordinated changes in gene expression of a large number of genes in the same pathway or function. This analysis does not set an arbitrary threshold of significance (like an overrepresentation analysis does) but instead considers all genes in the analysis. Then, the gene-level statistics from the inputted dataset are considered together to create one pathway-level statistic. This analysis can be particularly helpful if the differential expression analysis generates only a small list of significantly differentially expressed genes.
```{r GSEA with ClusterProfiler}
gseaKEGG <- gseKEGG(geneList = foldchanges, #inputting an ordered and named vector of fold changes with Entrez IDs as the associated names
              organism = "mmu", #mmu is mouse
              nPerm = 10000, 
              minGSSize = 15,
              pvalueCutoff = 0.15, #padj cutoff value
              pAdjustMethod = "BH",
              verbose = FALSE)

## Extract the GSEA results
gseaKEGG_results <- gseaKEGG@result
#Export CSV of GSEA results
write.csv(gseaKEGG_results, "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/GSEA/MatLiv_MDvEtOH_GSEA_KEGG_resultstable.csv")
```
```{r GSEA table summary, echo = FALSE}
print(head(gseaKEGG_results))
```

### GSEA output notes
I decided to set the p-value at 0.15 to capture a few of the pathways that will be significant sometimes and then not significant at other times, such as carbon metabolism, the TCA cycle, and the pentose phosphate pathway.

One annoying thing about this analysis is that it appears to change the significance level and p-value of certain pathways depending on when I run it. However, the same pathways/terms show up approximately where they used to be in the list order, albeit with slightly different p-values and order. Thus, do not pay much attention to the order of terms on the list or the exact p-values, but pay more attention to the relative level of each term. See below (Column 6). I plan to complete other analyses using other packages to confirm the data.

### GSEA output results
This analysis generates a table of the significantly differentially expressed pathways. 
* **Column 1** *(ID)* is the KEGG ID, which can be used to find the pathway. 
* **Column 2** *(Description)* describes the KEGG pathway and what it does. 
* **Column 3** *(setSize)* is the total number of genes annotated to that KEGG pathway. 
* **Column 4** *(enrichmentScore)* shows "the degree to which a [gene] set S is over-represented at the top or bottom of the ranked list L. The score is calculated by walking down the list L, increasing a running-sum statistic when we encounter a gene in S and decreasing when it is not. The magnitude of the increment depends on the gene statistics (e.g., correlation of the gene with phenotype). The ES is the maximum deviation from zero encountered in the random walk; it corresponds to a weighted Kolmogorov-Smirnov-like statistic(Subramanian et al. 2005)." (Quote from [here](https://yulab-smu.github.io/clusterProfiler-book/chapter2.html)). Thus, a more negative score means that the pathway was enriched for lower expressed genes, while a more positve score indicates that the pathway was enriched for genes with higher expression. A larger number indicates a higher enrichment.
* **Column 5** *(NES)* is the normalized enrichment score (NES), which is where the enrichment score is adjusted to account for differences in the sizes of the gene sets and the correlation between the gene sets and the expression data set. The NES allow comparison between the analysis results of different gene sets.
* **Column 6** *(pvalue)* is calculated using the permutation test, where the algorithm permutes the gene labels from the gene list L and recomputes the enrichment score of the gene set for the permuted data to create a null distribution for the enrichment score. Then, the p-value of the observed enrichment score is calculated relative to the null distribution. This method of permutation is why runs of the code even right after each other can generate different results. 
* **Column 7** *(p.adjust)* are the adjusted p-value using the Benjamini-Hochberg method. 
* **Column 8** *(qvalues)* are the FDR-corrected values, which are pretty close to the adjusted p-values, but slightly more liberal.
* **Column 9** *(rank)* unclear what this is. 
* **Column 10** *(leading_edge)* is the genes that appear in the ranked list "at or before the point where the running sum reaches its maximum deviation from zero. The leading-edge subset can be interpreted as the core that accounts for the gene setâ€™s enrichment sign ([source](http://software.broadinstitute.org/gsea/doc/GSEAUserGuideTEXT.htm#_Running_a_Leading_Edge%20Analysis)).
*  **Column 11** *(core_enrichment)* is the list of genes that had higher log2fold changes within the specific KEGG pathway. This is not all genes from the pathway that were in the input gene set, just the ones that had more extreme log2fold changes. 

### Visualize KEGG pathways colored by the fold change of the significant genes
This analysis uses an R package called [Pathview](https://rdrr.io/rforge/pathview/man/pathview.html) which is used to visualize the KEGG pathways with the up- and down-regulated genes highlighted for simpler viewing.
```{r Visualize KEGG pathways from GSEA, message = FALSE, warning = FALSE, eval = FALSE}
pathwayID <- gseaKEGG_results$ID
#This code will directly output the figures into the working directory. I will not print any of the pictures in this file due to space limitations. 
#Native KEGG view graphs
paths_KEGG <- pathview(gene.data = foldchanges,
              pathway.id = pathwayID,
              species = "mmu",
              limit = list(gene = 2, # value gives the max/min limit for foldchanges
              cpd = 1))
#Graphviz view graphs
paths_graphviz <- pathview(gene.data = foldchanges,
              pathway.id = pathwayID,
              species = "mmu",
              limit = list(gene = 2, # value gives the max/min limit for foldchanges
              cpd = 1),
              kegg.native = FALSE)
```

#### Interpretation of results of KEGG pathway visualization
The visualization above generates 2 different types of graphs: native KEGG view graphs and Graphviz view graphs, and both graphs are printed to the working directory. It also creates a third type of graph, which is not annotated with any inputted genes, but simply shows the normal state for the pathway. In both graphs, genes are colored with gray if their fold change is relatively close to zero, while they are colored more red (upregulated) or green (downregulated) depending upon their fold change. This analysis does **NOT** consider significance for the individual genes. It only considers fold change, and then provides an overall pathway significance value. What we are looking for is a coordinated change in a set of genes in a certain direction, which could indicate a functional change in the genes in response to alcohol. It is important to note that just because a pathway is significant does not mean that it is relevant, and the tissue under study should be considered (ex. a pathway implicated in gut barrier function is unlikely to perform the same function in the liver). At this point, biological knowledge must be integrated with the information provided from the code. 

## Signaling Pathway Impact Analysis (SPIA)
In contrast to the previous methods of overexpression analysis and gene set enrichment analysis, this method is one type of pathway topology analysis. This type of analysis considers gene interaction information in addition to the adjusted p-values and fold changes to identify pathways that are dysregulated. These analyses can explore how genes interact with one another (ex. activation, inhibition, p-lation, etc.) to create pathway-level statistics. [SPIA](http://bioconductor.org/packages/release/bioc/manuals/SPIA/man/SPIA.pdf) integrates this pathway information with p-values and log fold changes to find significantly differentially regulated pathways.
```{r SPIA analysis, message = FALSE, warning = FALSE, eval = FALSE}
##The significant gene list is a vector of fold changes where the names are ENTREZ gene IDs. The background set is a vector of all the genes represented on the platform.
all_genes <- as.character(file_entrez$entrez) #create background gene list of all labeled genes
sig_res_entrez <- file_entrez[which(file_entrez$padj < 0.1), ]
sig_entrez <- sig_res_entrez$log2FoldChange
names(sig_entrez) <- sig_res_entrez$entrez

#Run SPIA analysis
library("SPIA")
spia_result <- spia(de=sig_entrez, all=all_genes, organism="mmu", nB = 2000, plots = TRUE)
write.csv(spia_result, "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/GSEA/MatLiv_MDvEtOH_SPIA_resultstable.csv")
```
```{r echo = FALSE, eval = FALSE}
print(head(spia_result))
```

```{r Plotting SPIA results, eval = FALSE}
#Print just the significant rows
subset(spia_result, ID == "04062")
#Plot SPIA results
plot <- plotP(spia_result, threshold=0.1)
```
```{r echo = FALSE}
print(plot)
```

### SPIA data interpretation
The main output of this analysis is a large table containing all information from the analysis, including pathways and functions that are not significant. Below is a list of what each column means:
* **Name**- name of the KEGG annotated pathway
* **ID**- ID of the KEGG pathway
* **pSize**- the total number of genes annotated to that pathway
* **NDE** - the number of differentially expressed genes (provided by the user) that are annotated to that pathway, regardless of the direction of the fold change
* **pNDE**- the probability of seeing NDE genes on the pathway [unclear what this means]
* **tA**- the total "perturbation accumulation" in the pathway. i.e. combination of differential expression and log fold change as well as accounting for the importance of players in the pathway
* **pPERT**- the probability of observing a total score more extreme than tA by chance (i.e. the overall p-value)
* **pG**- a combined p-value from pNDE and pPERT
* **pGFdr**- the False Discovery Rate- adjusted global p-values
* **pGFWER**- the Bonferroni-adjusted global p-values
* **Status**- shows the direction in which the pathway is perturbed (either activated or inhibited)
* **KEGGLINK**- the web link to the KEGG website to view the pathway image with the differentially expressed genes highlighted in red. There is no indication on this pathway whether they are up or down, but the "status" will give an idea of what is happening. 

For the SPIA plot, each pathway is represented a point, with the coordinates determined as the log of the pNDE (p-value of the differentially expressed genes; using a hypergeo-metric model) and the p-value from perturbations (pPERT). The dotted diagonal lines show thesignificance regions based on the data.

## KEGG Enrichment
```{r Trying out a KEGG enrichment}
all_genes <- as.character(file_entrez$entrez) #create background gene list of all labeled genes
sig_genes <- file_entrez[which(file_entrez$padj < 0.1), ] #select out significantly expressed genes
sig_gene_list <- as.character(sig_genes$entrez) #create list of all significantly expressed genes
eKEGG <- enrichKEGG(gene = sig_gene_list,
                    organism = "mmu",
                    keyType = "kegg",
                    pvalueCutoff = 0.1,
                    universe = all_genes,
                    minGSSize = 10, maxGSSize = 500)
## Extract the KEGG enrichment results
eKEGG_results <- eKEGG@result
```
```{r echo = FALSE}
print(head(eKEGG_results))
```

This analysis did not yield any significant KEGG pathways with overrepresented genes assigned (p < 0.1). The lowest p-value after adjustment was nearly 0.5. Thus, I did not proceed with this analysis. Enrichment for KEGG modules also did not yield any significant resutls (this time, the lowest adjusted p-value was nearly 0.8). 

## Disease Ontology Analysis
This analysis is like normal gene ontology overexpression analysis, but it matches to disease ontologies. For example, a set of genes could be altered in breast cancer, and this set of genes could be found in a gene set of interest. 
```{r Disease ontology analysis}
enrichDO <- enrichDO(gene = sig_gene_list,
              ont = "DO",
              pvalueCutoff  = 0.1,
              pAdjustMethod = "BH",
              universe      = all_genes,
              minGSSize     = 5,
              maxGSSize     = 500,
              qvalueCutoff  = 0.2,
              readable      = FALSE)
enrichDO <- setReadable(enrichDO, 'org.Mm.eg.db')
enrichDO_table <- enrichDO@result
```

This approach did not yield any significant results. The lowest adjusted p-value was 0.38, and no more than 2 genes annotated to any single disease term, so this analysis was not useful for our purposes. I will not proceed with this analysis.

# GAGE Analysis
GAGE, or Generally Applicable Gene-set Enrichment, can draw conclusions about gene sets/pathways (since I use KEGG in this analysis, the conclusions are about pathways) that are dysregulated relative to a conrol gene set. Although GAGE was initially made for microarray data, it can also be used on RNAseq data. I originally tried this analysis to confirm or add to the results generated by the previous KEGG analysis using the ClusterProfiler package. There are some reports that it is too liberal with the data, but I actually found fewer results from this analysis than from the previous KEGG analysis (as will be apparent). 
```{r}
library(gage)
library(pathview)
kg.mmu=kegg.gsets(species = "mmu", id.type = "entrez") #create reference list by downloading directly from KEGG
kegg.gs=kg.mmu$kg.sets[kg.mmu$sigmet.idx] #create a gene list from the total KEGG list

#create large matrix just with the Entrez ID and the sample read numbers
gage_file <- file_entrez[,c(3,15:30)]
gage_file_unnamed <- gage_file[,-1]
rownames(gage_file_unnamed) <- gage_file[,1]
gage_matrix <- as.matrix(gage_file_unnamed)

ref_index=1:8
samp_index=9:16

#Run GAGE with ability to consider genes in the pathway that go in opposite directions from the same pathway
gage_kegg_bothdir <- gage(gage_matrix, 
                    gsets = kegg.gs, 
                    ref = ref_index,
                    samp = samp_index, 
                    et.size = c(10, 500),
                    same.dir = FALSE, #this sets the code to consider pathways where genes might go up or down
                    compare ="as.group")
#Create table from results
gage_greater_table <- gage_kegg_bothdir[["greater"]]
write.csv(gage_greater_table, "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/GSEA/MatLiv_MDvEtOH_GAGE_KEGG_consideringbothdir_resultstable.csv")
```
```{r GAGE both table summary, echo = FALSE}
print(head(gage_greater_table))
```
```{r}
#Run GAGE with genes separated into whether alcohol makes them go up or down
gage_kegg_eachdir <- gage(gage_matrix, 
                    gsets = kegg.gs, 
                    ref = ref_index,
                    samp = samp_index, 
                    et.size = c(10, 500),
                    same.dir = TRUE, #this sets code to only consider a pathway if all genes inside go either up or down
                    compare ="as.group")
#Create tables from results
gage_eachdir_greater_table <- gage_kegg_eachdir[["greater"]]
write.csv(gage_eachdir_greater_table, "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/GSEA/MatLiv_MDvEtOH_GAGE_KEGG_onlyup_resultstable.csv")
```
```{r GAGE up table summary, echo = FALSE}
print(head(gage_eachdir_greater_table))
```
```{r}
#Create tables from results
gage_eachdir_lesser_table <- gage_kegg_eachdir[["less"]]
write.csv(gage_eachdir_lesser_table, "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/GSEA/MatLiv_MDvEtOH_GAGE_KEGG_onlydown_resultstable.csv")
```
```{r GAGE down table summary, echo = FALSE}
print(head(gage_eachdir_lesser_table))
```
```{r eval = FALSE}
#This code is not run. It can only be run if the "1ongroup" option is used for the "compare" command.
#This code generates a figure for the top 10 KEGG pathways that are altered by alcohol
gage_matrix_red = gage_matrix[, samp_index]-rowMeans(gage_matrix[, ref_index]) #this subtracts the individual alcohol gene changes from the mean of the MD fold changes
select <- gage_kegg_bothdir$greater[, "p.val"] < 0.2 & !is.na(gage_kegg_bothdir$greater[,"q.val"]) #this selects out the KEGG pathways with a p-value less than 0.2 and with a calculated q-value
path.ids <- rownames(gage_kegg_bothdir$greater)[select] #this adds the rownames to a new variable
path.ids2 <- substr(path.ids, 1, 8) #this subsets the alcohol samples
pv.out.list <- sapply(path.ids2[1:10], function(pid) pathview(gene.data = gage_matrix_red, pathway.id = pid, species = "mmu", limit = list(gene = 4, cpd = 1))) #this views the KEGG pathway of the individual alcohol samples
```

Using the argument "same.dir = TRUE" means that for a given pathway, **ALL** of the genes must either go up or down, and then whether they go up or down goes into a separate part of the final variable, either "greater" or "less". In contrast, using the argument "same.dir = FALSE" means that the algorithm is just looking for a lot of genes changing in a particular pathway, and it allows genes within the same pathway to go both up and down.

I went back and forth as to what parameter to set for the "compare" argument under the "gage" command. The two options I was considering were "as.group" and "1ongroup". "As.group" compares the two groups as a whole, while "1ongroup" compares the individual alcohol individuals to the MD control as a group. I decided to use the "as.group" command since this is how we conducted the previous analyses. The code to graph the results of the "1ongroup" is deactivated, but can be reactivated or run as normal if desired. This is more informative if you want to see how each gene changes individually within a treatment group. This analysis also might be useful if you want to see the impact of removing or adding in certain individual outliers. In summary, GAGE analysis did not yield anything intereting to me at this point in time, but it could be very useful in the future to examine individual comparisons.

# Session Information
```{r, echo = FALSE}
session_info <- devtools::session_info()
print(session_info)
```