---
title: "Overexpression gProfileR Analysis of Maternal Liver Data"
author: "Kaylee Helfrich"
date: "7/26/2019"
output: 
  pdf_document: default
  html_document: default
data: DEG_Results_MDvAlc_MatLiv_LFCShrink_VSTParam_RemovedNA_ExcelProof.txt
---

Built with R `r getRversion()`
```{r RMarkdown setup, include = FALSE, message = FALSE}
require(knitr)
knitr::opts_chunk$set(echo = TRUE, tidy.opts=list(width.cutoff=60), tidy=TRUE)
knitr::opts_knit$set(root.dir = "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/gProfiler")
options(tinytex.verbose = TRUE)
```

# gProfileR Analysis of Maternal Liver Transcriptome Data
This document details my code and output for the gProfileR on maternal liver transcriptome data. Litter #'s: A1,A2,A3,A4,A5,A6,A7,A8,B1,B2,B4,B5,B6,B7,B8,B9 (where A is MD and B is Alc). 3g/kg alcohol as oral gavage (100% MD, split dose) from E8.5-17.5. Generated by KH, CK, NS Apr-June 2018.

```{r include = FALSE}
rm(list=ls())
setwd("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/gProfiler")
library(gProfileR)
library(treemap)
```

```{r Load file, message = FALSE, warning = FALSE}
file <- read.table("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/gProfiler/DEG_Results_MDvAlc_MatLiv_LFCShrink_VSTParam_RemovedNA_ExcelProof.txt", header = TRUE, sep = "")
file <- file[,-1] #remove first column which has meaningless numbers
```

# gProfileR Analysis
This analysis is similar to the analysis done by clusterProfiler, but this analysis also considers the ranking of the p-values of the genes, and the ones with lower p-values are considered more strongly. gProfileR also considers things other than Gene Ontology terms (whereas clusterProfiler only considers GO), such as biological pathways, regulatory motifs of transcription factors and microRNAs, human disease annotations, and protein-protein interactions. According to [this site](https://hbctraining.github.io/DGE_workshop/lessons/functional_analysis_other_methods.html), the output of GO terms from gprofileR are similar to the ones from clusterProfiler, but there will be differences due to differences in the algorithms used by the tools. It is possible to allow functional annotations that have been determined using *in silico* methods to suggest functions from electronic annotation. This information is of lower confidence than experimental and computational studies, and can be exluded using "exclude_iea = T".

Once this code has been run once, you can run only the parts to regenerate the graph (under command "treemap"). However, if you want to change parameters of how the data was produced originally, then you will need to run the entire code once again. 
```{r prepare data for gprofiler}
#create ranked gene ID list in ascending order of p-value (must be ascending for the analysis to consider the significance of the p-values)
sortedIDs <- file[order(file$padj),] #sort the p-values
sig_ps_list <- sortedIDs[which(sortedIDs$padj < 0.1), ] #subset out the p-values less than 0.1
sortedID_list <- as.character(sig_ps_list$GeneID) #just get the GeneIDs in a list
backgroundIDs <- as.character(file$GeneID) #create a background list of total genes measured
```

```{r gprofiler on all significant genes}
gprofiler_results_oe <- gprofiler(query = sortedID_list, 
                                  organism = "mmusculus",
                                  ordered_query = T, #telling it you provided an ordered p-value list
                                  significant = T, #just provide significant terms
                                  exclude_iea = F, #allow it to include terms not experimentally tested
                                  max_p_value = 0.1, 
                                  max_set_size = 0,
                                  correction_method = "fdr",
                                  hier_filtering = "none", 
                                  domain_size = "annotated",
                                  custom_bg = backgroundIDs)
#Subset and reorder gProfiler results to only include columns of interest
gprofiler_results_oe_reordered <- gprofiler_results_oe[, c("term.id", "domain", "term.name", "p.value", "overlap.size", "term.size", "intersection")]
## Order the results by p-adjusted value
gprofiler_results_oe_reordered <- gprofiler_results_oe_reordered[order(gprofiler_results_oe_reordered$p.value), ]
#Export CSV of all results
write.csv(gprofiler_results_oe_reordered, "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/gProfiler/MatLiv_MDvEtOH_gProfiler_AllSigGenes_resultstable.csv")
```
```{r gProfiler table summary, echo = FALSE}
print(head(gprofiler_results_oe_reordered))
```
```{r organize table for REVIGO}
## Extract only the 'GO' terms from the results. Allows for treemap
gprofiler_results_oe_GOs <- gprofiler_results_oe_reordered[grep('GO:', gprofiler_results_oe_reordered$term.id), ]
## Write the enriched GO results to file
write.csv(gprofiler_results_oe_GOs, "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/gProfiler/MatLiv_MDvAlc_Gprofiler_Allgenes_GOresults.csv")
```
```{r}
GOpval_oe <- gprofiler_results_oe_GOs[ , c("term.id", "p.value")]
write.table(GOpval_oe, "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/gProfiler/MatLiv_MDvAlc_Gprofiler_Allgenes_GOresults_simple.txt", quote=FALSE, row.names = FALSE, col.names = FALSE)
```

The code below (ommitted from being printed to the RMarkdown file) was produced via [REVIGO](http://revigo.irb.hr/). REVIGO is a tool accessed via the internet that takes in a list of GO terms, summarizes redundant terms, and graphs the result. To generate this, you input your genes as a list (as 2 columns with no labels with the GO term on the left and the p-value on the right; this is generated in the previous code chunk) into REVIGO and hit submit. Once it finishes running, then click on the "Treemap" tab. At the bottom of this page is a link to an R script. Open this, and then copy and paste into the script. Run this information to generate the graph in R. Edit bottom of script as needed in this R file. For all data used here, I submitted the analysis with default REVIGO settings.

## REVIGO output for graph
All data below was produced from the REVIGO website. 
```{r echo = FALSE}
revigo.names <- c("term_ID","description","freqInDbPercent","abslog10pvalue","uniqueness","dispensability","representative");
revigo.data <- rbind(c("GO:0002376","immune system process",0.600,1.6216,0.992,0.000,"immune system process"),
c("GO:0032502","developmental process",2.812,1.3883,0.992,0.000,"developmental process"),
c("GO:0035458","cellular response to interferon-beta",0.003,3.1851,0.859,0.000,"cellular response to interferon-beta"),
c("GO:1903201","regulation of oxidative stress-induced cell death",0.011,1.3197,0.723,0.676,"cellular response to interferon-beta"),
c("GO:1901700","response to oxygen-containing compound",0.503,1.9788,0.865,0.429,"cellular response to interferon-beta"),
c("GO:1902229","regulation of intrinsic apoptotic signaling pathway in response to DNA damage",0.007,1.3893,0.750,0.367,"cellular response to interferon-beta"),
c("GO:0051716","cellular response to stimulus",9.561,1.7077,0.824,0.680,"cellular response to interferon-beta"),
c("GO:0035456","response to interferon-beta",0.004,2.0773,0.885,0.640,"cellular response to interferon-beta"),
c("GO:0051707","response to other organism",0.299,1.9136,0.888,0.365,"cellular response to interferon-beta"),
c("GO:0002495","antigen processing and presentation of peptide antigen via MHC class II",0.004,1.5751,0.966,0.608,"cellular response to interferon-beta"),
c("GO:1901796","regulation of signal transduction by p53 class mediator",0.012,1.3270,0.805,0.558,"cellular response to interferon-beta"),
c("GO:0044565","dendritic cell proliferation",0.000,1.5817,0.850,0.541,"cellular response to interferon-beta"),
c("GO:0035902","response to immobilization stress",0.001,1.6536,0.919,0.169,"cellular response to interferon-beta"),
c("GO:0010038","response to metal ion",0.132,1.3197,0.874,0.566,"cellular response to interferon-beta"),
c("GO:0043627","response to estrogen",0.010,1.3197,0.919,0.274,"cellular response to interferon-beta"),
c("GO:0009607","response to biotic stimulus",0.342,1.7328,0.900,0.370,"cellular response to interferon-beta"),
c("GO:0009605","response to external stimulus",1.370,1.9136,0.889,0.251,"cellular response to interferon-beta"),
c("GO:0042221","response to chemical",3.071,1.5867,0.881,0.475,"cellular response to interferon-beta"),
c("GO:0070493","thrombin-activated receptor signaling pathway",0.005,1.3197,0.827,0.259,"cellular response to interferon-beta"),
c("GO:0009725","response to hormone",0.335,1.8416,0.850,0.614,"cellular response to interferon-beta"),
c("GO:1990267","response to transition metal nanoparticle",0.000,1.3197,0.918,0.268,"cellular response to interferon-beta"),
c("GO:0009719","response to endogenous stimulus",0.526,1.3197,0.897,0.387,"cellular response to interferon-beta"),
c("GO:0045087","innate immune response",0.148,1.9031,0.872,0.342,"cellular response to interferon-beta"),
c("GO:0007584","response to nutrient",0.026,1.3969,0.880,0.602,"cellular response to interferon-beta"),
c("GO:0033194","response to hydroperoxide",0.003,1.6003,0.885,0.347,"cellular response to interferon-beta"),
c("GO:0042770","signal transduction in response to DNA damage",0.023,1.3893,0.800,0.408,"cellular response to interferon-beta"),
c("GO:0006952","defense response",0.568,1.6536,0.884,0.504,"cellular response to interferon-beta"),
c("GO:0006950","response to stress",4.575,1.3969,0.877,0.562,"cellular response to interferon-beta"),
c("GO:0038026","reelin-mediated signaling pathway",0.002,1.6536,0.834,0.200,"cellular response to interferon-beta"),
c("GO:0050896","response to stimulus",12.210,3.1851,0.993,0.000,"response to stimulus"),
c("GO:0061615","glycolytic process through fructose-6-phosphate",0.061,2.5243,0.515,0.018,"glycolytic process through fructose-6-phosphate"),
c("GO:1901292","nucleoside phosphate catabolic process",0.187,1.9136,0.595,0.546,"glycolytic process through fructose-6-phosphate"),
c("GO:0032787","monocarboxylic acid metabolic process",2.485,2.0048,0.651,0.399,"glycolytic process through fructose-6-phosphate"),
c("GO:0055086","nucleobase-containing small molecule metabolic process",4.917,1.3565,0.641,0.478,"glycolytic process through fructose-6-phosphate"),
c("GO:0006775","fat-soluble vitamin metabolic process",0.012,1.3893,0.806,0.195,"glycolytic process through fructose-6-phosphate"),
c("GO:0006796","phosphate-containing compound metabolic process",13.110,1.4815,0.710,0.504,"glycolytic process through fructose-6-phosphate"),
c("GO:0034404","nucleobase-containing small molecule biosynthetic process",1.408,1.6925,0.562,0.629,"glycolytic process through fructose-6-phosphate"),
c("GO:0030388","fructose 1,6-bisphosphate metabolic process",0.014,1.8182,0.783,0.321,"glycolytic process through fructose-6-phosphate"),
c("GO:0046365","monosaccharide catabolic process",0.115,1.6536,0.715,0.679,"glycolytic process through fructose-6-phosphate"),
c("GO:0016052","carbohydrate catabolic process",1.078,1.9136,0.868,0.492,"glycolytic process through fructose-6-phosphate"),
c("GO:0010677","negative regulation of cellular carbohydrate metabolic process",0.015,1.6536,0.837,0.345,"glycolytic process through fructose-6-phosphate"),
c("GO:0071900","regulation of protein serine/threonine kinase activity",0.179,1.3575,0.750,0.343,"glycolytic process through fructose-6-phosphate"),
c("GO:0046434","organophosphate catabolic process",0.365,1.5884,0.702,0.579,"glycolytic process through fructose-6-phosphate"),
c("GO:0009068","aspartate family amino acid catabolic process",0.072,1.3893,0.700,0.499,"glycolytic process through fructose-6-phosphate"),
c("GO:0044262","cellular carbohydrate metabolic process",1.257,1.5391,0.874,0.627,"glycolytic process through fructose-6-phosphate"),
c("GO:0046939","nucleotide phosphorylation",0.792,2.0048,0.569,0.498,"glycolytic process through fructose-6-phosphate"),
c("GO:1901137","carbohydrate derivative biosynthetic process",3.651,2.0773,0.830,0.483,"glycolytic process through fructose-6-phosphate"),
c("GO:0009132","nucleoside diphosphate metabolic process",0.698,1.9281,0.574,0.619,"glycolytic process through fructose-6-phosphate"),
c("GO:0009152","purine ribonucleotide biosynthetic process",1.337,1.9914,0.505,0.679,"glycolytic process through fructose-6-phosphate"),
c("GO:0019320","hexose catabolic process",0.043,1.7721,0.730,0.635,"glycolytic process through fructose-6-phosphate"),
c("GO:0006734","NADH metabolic process",0.007,1.6840,0.657,0.520,"glycolytic process through fructose-6-phosphate"),
c("GO:0003351","epithelial cilium movement",0.005,1.3893,0.838,0.107,"glycolytic process through fructose-6-phosphate"),
c("GO:0061732","mitochondrial acetyl-CoA biosynthetic process from pyruvate",0.001,1.6308,0.657,0.492,"glycolytic process through fructose-6-phosphate"),
c("GO:1902750","negative regulation of cell cycle G2/M phase transition",0.027,1.6459,0.786,0.462,"glycolytic process through fructose-6-phosphate"),
c("GO:0006066","alcohol metabolic process",0.422,1.5817,0.734,0.249,"glycolytic process through fructose-6-phosphate"),
c("GO:0006532","aspartate biosynthetic process",0.002,1.3197,0.754,0.524,"glycolytic process through fructose-6-phosphate"),
c("GO:0061624","fructose catabolic process to hydroxyacetone phosphate and glyceraldehyde-3-phosphate",0.000,1.6536,0.733,0.444,"glycolytic process through fructose-6-phosphate"),
c("GO:0006116","NADH oxidation",0.002,1.6536,0.677,0.566,"glycolytic process through fructose-6-phosphate"),
c("GO:0006106","fumarate metabolic process",0.049,1.6635,0.738,0.295,"glycolytic process through fructose-6-phosphate"),
c("GO:0006090","pyruvate metabolic process",0.817,2.0773,0.666,0.590,"glycolytic process through fructose-6-phosphate"),
c("GO:0009890","negative regulation of biosynthetic process",0.772,1.3197,0.831,0.632,"glycolytic process through fructose-6-phosphate"),
c("GO:1902532","negative regulation of intracellular signal transduction",0.128,1.3197,0.753,0.667,"glycolytic process through fructose-6-phosphate"),
c("GO:0042866","pyruvate biosynthetic process",0.005,2.0048,0.737,0.549,"glycolytic process through fructose-6-phosphate"),
c("GO:0044282","small molecule catabolic process",1.273,1.3830,0.692,0.674,"glycolytic process through fructose-6-phosphate"),
c("GO:0072521","purine-containing compound metabolic process",2.673,1.8794,0.826,0.357,"glycolytic process through fructose-6-phosphate"),
c("GO:0072524","pyridine-containing compound metabolic process",1.351,2.0000,0.837,0.241,"glycolytic process through fructose-6-phosphate"),
c("GO:0005996","monosaccharide metabolic process",0.817,1.6536,0.696,0.598,"glycolytic process through fructose-6-phosphate"),
c("GO:0006001","fructose catabolic process",0.000,1.6536,0.805,0.593,"glycolytic process through fructose-6-phosphate"),
c("GO:0051900","regulation of mitochondrial depolarization",0.003,1.9393,0.913,0.018,"regulation of mitochondrial depolarization"),
c("GO:0071711","basement membrane organization",0.008,1.3261,0.818,0.266,"regulation of mitochondrial depolarization"),
c("GO:1901888","regulation of cell junction assembly",0.014,1.8327,0.765,0.142,"regulation of mitochondrial depolarization"),
c("GO:0051639","actin filament network formation",0.003,1.3893,0.846,0.252,"regulation of mitochondrial depolarization"),
c("GO:1904059","regulation of locomotor rhythm",0.001,1.5391,0.941,0.118,"regulation of mitochondrial depolarization"),
c("GO:1904060","negative regulation of locomotor rhythm",0.000,1.5391,0.930,0.687,"regulation of mitochondrial depolarization"),
c("GO:2000489","regulation of hepatic stellate cell activation",0.000,1.3893,0.842,0.119,"regulation of mitochondrial depolarization"),
c("GO:1901615","organic hydroxy compound metabolic process",0.831,1.3575,0.948,0.037,"organic hydroxy compound metabolism"),
c("GO:0034436","glycoprotein transport",0.001,1.6536,0.923,0.042,"glycoprotein transport"),
c("GO:0072337","modified amino acid transport",0.042,1.3778,0.972,0.286,"glycoprotein transport"),
c("GO:0006820","anion transport",1.956,1.3036,0.970,0.413,"glycoprotein transport"),
c("GO:0071702","organic substance transport",4.980,1.3125,0.970,0.189,"glycoprotein transport"),
c("GO:0015881","creatine transport",0.000,1.4306,0.921,0.394,"glycoprotein transport"),
c("GO:0010172","embryonic body morphogenesis",0.005,1.9788,0.817,0.047,"embryonic body morphogenesis"),
c("GO:0098868","bone growth",0.005,1.5768,0.832,0.398,"embryonic body morphogenesis"),
c("GO:0010171","body morphogenesis",0.015,1.3125,0.840,0.440,"embryonic body morphogenesis"),
c("GO:0060485","mesenchyme development",0.056,1.3969,0.793,0.628,"embryonic body morphogenesis"),
c("GO:0010632","regulation of epithelial cell migration",0.036,1.5243,0.715,0.601,"embryonic body morphogenesis"),
c("GO:0001503","ossification",0.074,1.3107,0.818,0.583,"embryonic body morphogenesis"),
c("GO:0001501","skeletal system development",0.107,1.4413,0.796,0.659,"embryonic body morphogenesis"),
c("GO:0034447","very-low-density lipoprotein particle clearance",0.001,1.5391,0.826,0.370,"embryonic body morphogenesis"),
c("GO:0030539","male genitalia development",0.006,1.3197,0.820,0.541,"embryonic body morphogenesis"),
c("GO:0003197","endocardial cushion development",0.009,1.7122,0.807,0.553,"embryonic body morphogenesis"),
c("GO:0002066","columnar/cuboidal epithelial cell development",0.031,1.3197,0.752,0.680,"embryonic body morphogenesis"),
c("GO:0021853","cerebral cortex GABAergic interneuron migration",0.001,1.5867,0.741,0.578,"embryonic body morphogenesis"),
c("GO:0097154","GABAergic neuron differentiation",0.003,1.3206,0.767,0.616,"embryonic body morphogenesis"),
c("GO:0001701","in utero embryonic development",0.056,1.6861,0.801,0.663,"embryonic body morphogenesis"),
c("GO:0007417","central nervous system development",0.191,1.9136,0.782,0.508,"embryonic body morphogenesis"),
c("GO:2001014","regulation of skeletal muscle cell differentiation",0.004,1.4815,0.741,0.527,"embryonic body morphogenesis"),
c("GO:0044272","sulfur compound biosynthetic process",1.235,1.9508,0.851,0.050,"sulfur compound biosynthesis"),
c("GO:0019217","regulation of fatty acid metabolic process",0.021,1.5817,0.685,0.524,"sulfur compound biosynthesis"),
c("GO:0019242","methylglyoxal biosynthetic process",0.012,1.3197,0.736,0.621,"sulfur compound biosynthesis"),
c("GO:0010565","regulation of cellular ketone metabolic process",0.049,1.3197,0.719,0.677,"sulfur compound biosynthesis"),
c("GO:0008610","lipid biosynthetic process",2.123,1.9393,0.707,0.191,"sulfur compound biosynthesis"),
c("GO:0008202","steroid metabolic process",0.161,1.3565,0.812,0.628,"sulfur compound biosynthesis"),
c("GO:0006641","triglyceride metabolic process",0.038,1.3089,0.792,0.550,"sulfur compound biosynthesis"),
c("GO:0006629","lipid metabolic process",3.522,1.5575,0.808,0.218,"sulfur compound biosynthesis"),
c("GO:0006082","organic acid metabolic process",9.086,1.3768,0.622,0.608,"sulfur compound biosynthesis"),
c("GO:0044283","small molecule biosynthetic process",5.677,1.9788,0.641,0.589,"sulfur compound biosynthesis"),
c("GO:0044281","small molecule metabolic process",15.138,1.3893,0.783,0.300,"sulfur compound biosynthesis"),
c("GO:1901135","carbohydrate derivative metabolic process",6.319,2.0048,0.936,0.059,"carbohydrate derivative metabolism"),
c("GO:0051186","cofactor metabolic process",3.985,1.3575,0.915,0.078,"cofactor metabolism"),
c("GO:0006793","phosphorus metabolic process",13.507,1.5317,0.903,0.114,"cofactor metabolism"),
c("GO:0005975","carbohydrate metabolic process",5.260,1.8962,0.928,0.079,"carbohydrate metabolism"));

stuff <- data.frame(revigo.data);
names(stuff) <- revigo.names;

stuff$abslog10pvalue <- as.numeric( as.character(stuff$abslog10pvalue) );
stuff$freqInDbPercent <- as.numeric( as.character(stuff$freqInDbPercent) );
stuff$uniqueness <- as.numeric( as.character(stuff$uniqueness) );
stuff$dispensability <- as.numeric( as.character(stuff$dispensability) );
```

### Treeplot for REVIGO data- all genes
```{r create treeplot for all significantly differentially expressed genes}
#prepare pdf for treeplot
pdf(file="~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/gProfiler/MatLiv_MDvAlc_allgenes_Revigo_treemap.pdf", width=16, height=9)
#create treeplot
treemap_all <- treemap(stuff,
	index = c("representative","description"),
	vSize = "abslog10pvalue",
	type = "categorical",
	vColor = "representative",
	title = "REVIGO Gene Ontology treemap",
	inflate.labels = FALSE, # set this to TRUE for space-filling group labels- good for posters
	lowerbound.cex.labels = 0, #draw as many labels as possible, but some squares may not get a label)
	bg.labels = "#CCCCCCAA", #define background color of group labels, "#CCCCCC00" is fully transparent, "#CCCCCCAA" is semi-transparent grey, NA is opaque
	position.legend = "none")
dev.off()
```

# gProfileR Analysis on up-regulated genes
```{r subset up-regulated genes}
sig_ps_up <- sig_ps_list[which(sig_ps_list$log2FoldChange > 0), ] #find genes that are up-regulated from the significant list
sig_ps_up_list <- as.character(sig_ps_up$GeneID) #turn those genes into a list of Gene IDs
```
```{r run gProfiler}
gprofiler_results_oe_up <- gprofiler(query = sig_ps_up_list, 
                                  organism = "mmusculus",
                                  ordered_query = T, 
                                  significant = T,
                                  exclude_iea = F, 
                                  max_p_value = 0.1, 
                                  max_set_size = 0,
                                  correction_method = "fdr",
                                  hier_filtering = "none", 
                                  domain_size = "annotated",
                                  custom_bg = backgroundIDs)
#Subset and reorder gProfiler results to only include columns of interest
gprofiler_results_oe_up_reordered <- gprofiler_results_oe_up[, c("term.id", "domain", "term.name", "p.value", "overlap.size", "term.size", "intersection")]
## Order the results by p-adjusted value
gprofiler_results_oe_up_reordered <- gprofiler_results_oe_up_reordered[order(gprofiler_results_oe_up_reordered$p.value), ]
#Export CSV of all results
write.csv(gprofiler_results_oe_up_reordered, "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/gProfiler/MatLiv_MDvEtOH_gProfiler_UpSigGenes_resultstable.csv")
```
```{r gProfiler up-genes table summary, echo = FALSE}
print(head(gprofiler_results_oe_up_reordered))
```
```{r organize up-genes table for downstream applications}
## Extract only the 'GO' terms from the results
gprofiler_results_oe_up_GOs <- gprofiler_results_oe_up_reordered[grep('GO:', gprofiler_results_oe_up_reordered$term.id), ]
## Write the enriched GO results to file
write.csv(gprofiler_results_oe_up_GOs, "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/gProfiler/MatLiv_MDvAlc_Gprofiler_Upgenes_GOresults.csv")
```
```{r}
GOpval_oe_up <- gprofiler_results_oe_up_GOs[ , c("term.id", "p.value")]
write.table(GOpval_oe_up, "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/gProfiler/MatLiv_MDvAlc_Gprofiler_Upgenes_GOresults_simple.txt", quote=FALSE, row.names = FALSE, col.names = FALSE)
```

## REVIGO output for graph
```{r echo = FALSE}
revigo.names <- c("term_ID","description","freqInDbPercent","abslog10pvalue","uniqueness","dispensability","representative");
revigo.data <- rbind(c("GO:0006820","anion transport",1.956,2.6253,0.943,0.000,"anion transport"),
c("GO:0006810","transport",17.616,1.3072,0.946,0.609,"anion transport"),
c("GO:0006811","ion transport",5.344,2.0173,0.950,0.417,"anion transport"),
c("GO:1901264","carbohydrate derivative transport",0.208,1.8356,0.880,0.283,"anion transport"),
c("GO:0035435","phosphate ion transmembrane transport",0.100,1.5654,0.947,0.680,"anion transport"),
c("GO:0034436","glycoprotein transport",0.001,1.8069,0.916,0.311,"anion transport"),
c("GO:0015711","organic anion transport",1.192,1.9666,0.936,0.663,"anion transport"),
c("GO:0007032","endosome organization",0.020,1.5317,0.956,0.000,"endosome organization"),
c("GO:2001286","regulation of caveolin-mediated endocytosis",0.001,1.5243,0.919,0.239,"endosome organization"),
c("GO:0010463","mesenchymal cell proliferation",0.009,1.3585,0.926,0.000,"mesenchymal cell proliferation"),
c("GO:0032502","developmental process",2.812,1.4365,0.991,0.000,"developmental process"),
c("GO:0035902","response to immobilization stress",0.001,1.8539,0.965,0.000,"response to immobilization stress"),
c("GO:1901700","response to oxygen-containing compound",0.503,1.5391,0.936,0.614,"response to immobilization stress"),
c("GO:0009628","response to abiotic stimulus",0.571,1.4045,0.955,0.340,"response to immobilization stress"),
c("GO:0043627","response to estrogen",0.010,1.4449,0.963,0.248,"response to immobilization stress"),
c("GO:0009743","response to carbohydrate",0.041,1.5654,0.933,0.692,"response to immobilization stress"),
c("GO:0009725","response to hormone",0.335,1.5884,0.923,0.224,"response to immobilization stress"),
c("GO:1990267","response to transition metal nanoparticle",0.000,1.4045,0.958,0.330,"response to immobilization stress"),
c("GO:0046686","response to cadmium ion",0.027,1.4045,0.945,0.485,"response to immobilization stress"),
c("GO:0040037","negative regulation of fibroblast growth factor receptor signaling pathway",0.003,1.4101,0.830,0.647,"response to immobilization stress"),
c("GO:0038026","reelin-mediated signaling pathway",0.002,1.8069,0.877,0.166,"response to immobilization stress"),
c("GO:0051179","localization",18.495,1.3072,0.993,0.000,"localization"),
c("GO:1901137","carbohydrate derivative biosynthetic process",3.651,4.0820,0.775,0.000,"carbohydrate derivative biosynthesis"),
c("GO:0030388","fructose 1,6-bisphosphate metabolic process",0.014,2.1013,0.724,0.424,"carbohydrate derivative biosynthesis"),
c("GO:0009100","glycoprotein metabolic process",0.356,1.8069,0.833,0.578,"carbohydrate derivative biosynthesis"),
c("GO:0006486","protein glycosylation",0.317,2.4685,0.690,0.571,"carbohydrate derivative biosynthesis"),
c("GO:1904059","regulation of locomotor rhythm",0.001,1.6696,0.951,0.000,"regulation of locomotor rhythm"),
c("GO:0055088","lipid homeostasis",0.041,1.5452,0.955,0.129,"regulation of locomotor rhythm"),
c("GO:0030811","regulation of nucleotide catabolic process",0.000,1.3372,0.653,0.547,"regulation of locomotor rhythm"),
c("GO:0045859","regulation of protein kinase activity",0.295,1.5482,0.672,0.454,"regulation of locomotor rhythm"),
c("GO:1904528","positive regulation of microtubule binding",0.001,1.5901,0.950,0.106,"regulation of locomotor rhythm"),
c("GO:0051902","negative regulation of mitochondrial depolarization",0.001,1.5243,0.916,0.531,"regulation of locomotor rhythm"),
c("GO:1904060","negative regulation of locomotor rhythm",0.000,1.6696,0.941,0.687,"regulation of locomotor rhythm"),
c("GO:0018196","peptidyl-asparagine modification",0.015,1.9508,0.906,0.037,"peptidyl-asparagine modification"),
c("GO:0035970","peptidyl-threonine dephosphorylation",0.003,1.4089,0.801,0.245,"peptidyl-asparagine modification"),
c("GO:0022417","protein maturation by protein folding",0.001,1.7100,0.934,0.137,"peptidyl-asparagine modification"),
c("GO:0006497","protein lipidation",0.128,1.6696,0.851,0.306,"peptidyl-asparagine modification"),
c("GO:0035383","thioester metabolic process",0.254,2.3152,0.881,0.048,"thioester metabolism"),
c("GO:0061732","mitochondrial acetyl-CoA biosynthetic process from pyruvate",0.001,2.3019,0.616,0.645,"thioester metabolism"),
c("GO:0035384","thioester biosynthetic process",0.151,2.5544,0.835,0.631,"thioester metabolism"),
c("GO:1901615","organic hydroxy compound metabolic process",0.831,1.4750,0.934,0.055,"organic hydroxy compound metabolism"),
c("GO:0016052","carbohydrate catabolic process",1.078,2.3019,0.814,0.057,"carbohydrate catabolism"),
c("GO:0016051","carbohydrate biosynthetic process",1.079,1.6696,0.673,0.627,"carbohydrate catabolism"),
c("GO:0010677","negative regulation of cellular carbohydrate metabolic process",0.015,1.8069,0.828,0.419,"carbohydrate catabolism"),
c("GO:0019242","methylglyoxal biosynthetic process",0.012,1.4295,0.734,0.416,"carbohydrate catabolism"),
c("GO:0009068","aspartate family amino acid catabolic process",0.072,1.5654,0.670,0.499,"carbohydrate catabolism"),
c("GO:0044262","cellular carbohydrate metabolic process",1.257,1.6696,0.842,0.627,"carbohydrate catabolism"),
c("GO:0006532","aspartate biosynthetic process",0.002,1.4045,0.739,0.524,"carbohydrate catabolism"),
c("GO:0061624","fructose catabolic process to hydroxyacetone phosphate and glyceraldehyde-3-phosphate",0.000,1.8069,0.682,0.312,"carbohydrate catabolism"),
c("GO:0006001","fructose catabolic process",0.000,1.8069,0.767,0.593,"carbohydrate catabolism"),
c("GO:0006000","fructose metabolic process",0.022,1.3747,0.725,0.442,"carbohydrate catabolism"),
c("GO:0032787","monocarboxylic acid metabolic process",2.485,2.5591,0.632,0.064,"monocarboxylic acid metabolism"),
c("GO:1901292","nucleoside phosphate catabolic process",0.187,2.4271,0.537,0.642,"monocarboxylic acid metabolism"),
c("GO:0033865","nucleoside bisphosphate metabolic process",0.239,1.8665,0.562,0.659,"monocarboxylic acid metabolism"),
c("GO:0055086","nucleobase-containing small molecule metabolic process",4.917,2.1701,0.622,0.518,"monocarboxylic acid metabolism"),
c("GO:0046365","monosaccharide catabolic process",0.115,1.8539,0.661,0.679,"monocarboxylic acid metabolism"),
c("GO:0046394","carboxylic acid biosynthetic process",4.159,2.3391,0.570,0.644,"monocarboxylic acid metabolism"),
c("GO:0044255","cellular lipid metabolic process",2.704,2.3152,0.712,0.256,"monocarboxylic acid metabolism"),
c("GO:0046434","organophosphate catabolic process",0.365,1.9626,0.635,0.596,"monocarboxylic acid metabolism"),
c("GO:0034033","purine nucleoside bisphosphate biosynthetic process",0.155,2.0947,0.528,0.661,"monocarboxylic acid metabolism"),
c("GO:0009117","nucleotide metabolic process",4.166,2.5544,0.445,0.466,"monocarboxylic acid metabolism"),
c("GO:0019320","hexose catabolic process",0.043,2.0362,0.678,0.635,"monocarboxylic acid metabolism"),
c("GO:0006734","NADH metabolic process",0.007,2.0306,0.616,0.474,"monocarboxylic acid metabolism"),
c("GO:0006066","alcohol metabolic process",0.422,1.7100,0.703,0.350,"monocarboxylic acid metabolism"),
c("GO:0000038","very long-chain fatty acid metabolic process",0.018,1.3788,0.715,0.536,"monocarboxylic acid metabolism"),
c("GO:0061615","glycolytic process through fructose-6-phosphate",0.061,2.8633,0.442,0.590,"monocarboxylic acid metabolism"),
c("GO:0006116","NADH oxidation",0.002,1.9101,0.639,0.566,"monocarboxylic acid metabolism"),
c("GO:0019471","4-hydroxyproline metabolic process",0.002,1.5607,0.770,0.307,"monocarboxylic acid metabolism"),
c("GO:0030497","fatty acid elongation",0.010,1.3335,0.695,0.546,"monocarboxylic acid metabolism"),
c("GO:0006106","fumarate metabolic process",0.049,1.9626,0.725,0.391,"monocarboxylic acid metabolism"),
c("GO:0006641","triglyceride metabolic process",0.038,1.4776,0.788,0.562,"monocarboxylic acid metabolism"),
c("GO:0006090","pyruvate metabolic process",0.817,2.6253,0.646,0.520,"monocarboxylic acid metabolism"),
c("GO:0006629","lipid metabolic process",3.522,2.1701,0.789,0.223,"monocarboxylic acid metabolism"),
c("GO:0006082","organic acid metabolic process",9.086,2.2541,0.602,0.592,"monocarboxylic acid metabolism"),
c("GO:0006072","glycerol-3-phosphate metabolic process",0.121,1.4789,0.676,0.529,"monocarboxylic acid metabolism"),
c("GO:0019637","organophosphate metabolic process",6.148,2.6253,0.672,0.580,"monocarboxylic acid metabolism"),
c("GO:0042866","pyruvate biosynthetic process",0.005,2.5591,0.720,0.324,"monocarboxylic acid metabolism"),
c("GO:0044283","small molecule biosynthetic process",5.677,2.0173,0.607,0.654,"monocarboxylic acid metabolism"),
c("GO:0044281","small molecule metabolic process",15.138,1.8633,0.766,0.300,"monocarboxylic acid metabolism"),
c("GO:0044282","small molecule catabolic process",1.273,1.5901,0.648,0.674,"monocarboxylic acid metabolism"),
c("GO:0072522","purine-containing compound biosynthetic process",1.502,2.5544,0.725,0.608,"monocarboxylic acid metabolism"),
c("GO:0070085","glycosylation",0.415,2.5544,0.837,0.171,"monocarboxylic acid metabolism"),
c("GO:0052646","alditol phosphate metabolic process",0.123,1.3883,0.680,0.530,"monocarboxylic acid metabolism"),
c("GO:0042761","very long-chain fatty acid biosynthetic process",0.009,1.4413,0.696,0.500,"monocarboxylic acid metabolism"),
c("GO:1901135","carbohydrate derivative metabolic process",6.319,4.0820,0.920,0.074,"carbohydrate derivative metabolism"),
c("GO:0072524","pyridine-containing compound metabolic process",1.351,2.5544,0.815,0.074,"pyridine-containing compound metabolism"),
c("GO:0072521","purine-containing compound metabolic process",2.673,2.1701,0.803,0.357,"pyridine-containing compound metabolism"),
c("GO:0042157","lipoprotein metabolic process",0.210,1.3028,0.934,0.075,"lipoprotein metabolism"),
c("GO:0006091","generation of precursor metabolites and energy",1.940,1.4365,0.910,0.078,"generation of precursor metabolites and energy"),
c("GO:0005975","carbohydrate metabolic process",5.260,2.3391,0.909,0.079,"carbohydrate metabolism"),
c("GO:0007417","central nervous system development",0.191,2.3565,0.800,0.082,"central nervous system development"),
c("GO:0098868","bone growth",0.005,1.7595,0.850,0.488,"central nervous system development"),
c("GO:0010172","embryonic body morphogenesis",0.005,1.4045,0.844,0.663,"central nervous system development"),
c("GO:0030879","mammary gland development",0.021,1.5751,0.826,0.586,"central nervous system development"),
c("GO:0034447","very-low-density lipoprotein particle clearance",0.001,1.6696,0.843,0.446,"central nervous system development"),
c("GO:1904936","interneuron migration",0.001,1.7122,0.769,0.588,"central nervous system development"),
c("GO:0003192","mitral valve formation",0.001,1.5654,0.844,0.470,"central nervous system development"),
c("GO:0021892","cerebral cortex GABAergic interneuron differentiation",0.002,1.5200,0.766,0.667,"central nervous system development"),
c("GO:0001755","neural crest cell migration",0.011,1.6198,0.739,0.560,"central nervous system development"),
c("GO:0009888","tissue development",0.491,1.3134,0.890,0.618,"central nervous system development"),
c("GO:0021830","interneuron migration from the subpallium to the cortex",0.002,1.5654,0.752,0.591,"central nervous system development"),
c("GO:0097154","GABAergic neuron differentiation",0.003,1.4295,0.781,0.616,"central nervous system development"),
c("GO:0001701","in utero embryonic development",0.056,1.7167,0.823,0.596,"central nervous system development"),
c("GO:0002651","positive regulation of tolerance induction to self antigen",0.000,1.3372,0.844,0.451,"central nervous system development"),
c("GO:0006732","coenzyme metabolic process",3.111,2.3307,0.813,0.083,"coenzyme metabolism"),
c("GO:0051186","cofactor metabolic process",3.985,1.6144,0.903,0.089,"cofactor metabolism"),
c("GO:0006793","phosphorus metabolic process",13.507,1.5702,0.889,0.114,"cofactor metabolism"));

stuff_up <- data.frame(revigo.data);
names(stuff_up) <- revigo.names;

stuff_up$abslog10pvalue <- as.numeric( as.character(stuff_up$abslog10pvalue) );
stuff_up$freqInDbPercent <- as.numeric( as.character(stuff_up$freqInDbPercent) );
stuff_up$uniqueness <- as.numeric( as.character(stuff_up$uniqueness) );
stuff_up$dispensability <- as.numeric( as.character(stuff_up$dispensability) );
```

### Treeplot for REVIGO data- overexpressed genes
```{r}
pdf(file="~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/gProfiler/MatLiv_MDvAlc_Upgenes_Revigo_treemap.pdf", width=16, height=9)
treemap_up <- treemap(stuff_up,
	index = c("representative","description"),
	vSize = "abslog10pvalue",
	type = "categorical",
	vColor = "representative",
	title = "REVIGO Gene Ontology treemap",
	inflate.labels = FALSE, #set this to TRUE for space-filling group labels - good for posters
	lowerbound.cex.labels = 0, #draw as many labels as possible, but some squares may not get a label)
	bg.labels = "#CCCCCCAA", #define background color of group labels, "#CCCCCC00" is fully transparent, "#CCCCCCAA" is semi-transparent grey, NA is opaque
	position.legend = "none")
dev.off()
```

# gProfileR Analysis on down-regulated genes
```{r}
sig_ps_down <- sig_ps_list[which(sig_ps_list$log2FoldChange < 0), ] #subset out the down-regulated genes
sig_ps_down_list <- as.character(sig_ps_down$GeneID) #create list of GeneIDs

#run gprofiler
gprofiler_results_oe_down <- gprofiler(query = sig_ps_down_list, 
                                  organism = "mmusculus",
                                  ordered_query = T, 
                                  significant = T,
                                  exclude_iea = F, 
                                  max_p_value = 0.1, 
                                  max_set_size = 0,
                                  correction_method = "fdr",
                                  hier_filtering = "none", 
                                  domain_size = "annotated",
                                  custom_bg = backgroundIDs)
#Subset and reorder gProfiler results to only include columns of interest
gprofiler_results_oe_down_reordered <- gprofiler_results_oe_down[, c("term.id", "domain", "term.name", "p.value", "overlap.size", "term.size", "intersection")]
## Order the results by p-adjusted value
gprofiler_results_oe_down_reordered <- gprofiler_results_oe_down_reordered[order(gprofiler_results_oe_down_reordered$p.value), ]
#Export CSV of all results
write.csv(gprofiler_results_oe_down_reordered, "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/gProfiler/MatLiv_MDvEtOH_gProfiler_DownSigGenes_resultstable.csv")
```
```{r gProfiler down-genes table summary, echo = FALSE}
print(head(gprofiler_results_oe_down_reordered))
```
```{r organize down-genes table for downstream applications}
## Extract only the 'GO' terms from the results
gprofiler_results_oe_down_GOs <- gprofiler_results_oe_down_reordered[grep('GO:', gprofiler_results_oe_down_reordered$term.id), ]
## Write the enriched GO results to file
write.csv(gprofiler_results_oe_down_GOs, "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/gProfiler/MatLiv_MDvAlc_Gprofiler_Downgenes_GOresults.csv")
```
```{r}
GOpval_oe_down <- gprofiler_results_oe_down_GOs[ , c("term.id", "p.value")]
write.table(GOpval_oe_down, "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/gProfiler/MatLiv_MDvAlc_Gprofiler_Downgenes_GOresults_simple.txt", quote=FALSE, row.names = FALSE, col.names = FALSE)
```

## REVIGO output for graph
```{r echo = FALSE}
revigo.names <- c("term_ID","description","freqInDbPercent","abslog10pvalue","uniqueness","dispensability","representative");
revigo.data <- rbind(c("GO:0001775","cell activation",0.171,4.5100,0.884,0.000,"cell activation"),
c("GO:0016049","cell growth",0.153,1.4609,0.882,0.145,"cell activation"),
c("GO:0006693","prostaglandin metabolic process",0.007,1.3507,0.879,0.398,"cell activation"),
c("GO:0006694","steroid biosynthetic process",0.122,1.6968,0.875,0.275,"cell activation"),
c("GO:0044108","cellular alcohol biosynthetic process",0.046,1.7100,0.847,0.133,"cell activation"),
c("GO:0003351","epithelial cilium movement",0.005,1.5735,0.864,0.114,"cell activation"),
c("GO:0003341","cilium movement",0.023,1.3706,0.852,0.635,"cell activation"),
c("GO:0006069","ethanol oxidation",0.017,1.4841,0.915,0.666,"cell activation"),
c("GO:0002376","immune system process",0.600,9.6003,0.994,0.000,"immune system process"),
c("GO:0003158","endothelium development",0.025,1.3605,0.944,0.000,"endothelium development"),
c("GO:0006044","N-acetylglucosamine metabolic process",0.055,2.7055,0.973,0.000,"N-acetylglucosamine metabolism"),
c("GO:0006040","amino sugar metabolic process",0.244,1.8182,0.973,0.383,"N-acetylglucosamine metabolism"),
c("GO:0006955","immune response",0.337,9.6003,0.655,0.000,"immune response"),
c("GO:0050920","regulation of chemotaxis",0.043,1.8268,0.596,0.661,"immune response"),
c("GO:1901699","cellular response to nitrogen compound",0.121,1.6055,0.761,0.594,"immune response"),
c("GO:1901700","response to oxygen-containing compound",0.503,1.4522,0.785,0.683,"immune response"),
c("GO:1902229","regulation of intrinsic apoptotic signaling pathway in response to DNA damage",0.007,2.3936,0.652,0.400,"immune response"),
c("GO:0050901","leukocyte tethering or rolling",0.004,1.8477,0.680,0.642,"immune response"),
c("GO:0051716","cellular response to stimulus",9.561,3.4698,0.756,0.680,"immune response"),
c("GO:0035458","cellular response to interferon-beta",0.003,7.1141,0.794,0.229,"immune response"),
c("GO:0035457","cellular response to interferon-alpha",0.002,1.5482,0.799,0.627,"immune response"),
c("GO:0035456","response to interferon-beta",0.004,5.5686,0.817,0.640,"immune response"),
c("GO:0051707","response to other organism",0.299,7.1141,0.779,0.321,"immune response"),
c("GO:0035455","response to interferon-alpha",0.003,1.7282,0.818,0.642,"immune response"),
c("GO:0089700","protein kinase D signaling",0.001,1.4711,0.752,0.211,"immune response"),
c("GO:0071281","cellular response to iron ion",0.005,1.7033,0.806,0.447,"immune response"),
c("GO:0048583","regulation of response to stimulus",1.120,1.3686,0.658,0.420,"immune response"),
c("GO:1903689","regulation of wound healing, spreading of epidermal cells",0.001,1.4486,0.616,0.665,"immune response"),
c("GO:0072376","protein activation cascade",0.018,1.4535,0.843,0.258,"immune response"),
c("GO:0002495","antigen processing and presentation of peptide antigen via MHC class II",0.004,3.3045,0.814,0.639,"immune response"),
c("GO:1901796","regulation of signal transduction by p53 class mediator",0.012,1.8182,0.676,0.603,"immune response"),
c("GO:0044565","dendritic cell proliferation",0.000,2.5686,0.745,0.566,"immune response"),
c("GO:0061756","leukocyte adhesion to vascular endothelial cell",0.004,1.6421,0.919,0.695,"immune response"),
c("GO:0070887","cellular response to chemical stimulus",1.007,3.0511,0.747,0.603,"immune response"),
c("GO:0009607","response to biotic stimulus",0.342,6.6615,0.836,0.340,"immune response"),
c("GO:0009605","response to external stimulus",1.370,5.3585,0.818,0.390,"immune response"),
c("GO:0033004","negative regulation of mast cell activation",0.002,1.3507,0.596,0.610,"immune response"),
c("GO:0042221","response to chemical",3.071,2.0768,0.806,0.562,"immune response"),
c("GO:0070493","thrombin-activated receptor signaling pathway",0.005,1.6968,0.731,0.295,"immune response"),
c("GO:0030522","intracellular receptor signaling pathway",0.075,1.8356,0.684,0.286,"immune response"),
c("GO:1903975","regulation of glial cell migration",0.001,1.6536,0.689,0.665,"immune response"),
c("GO:0034097","response to cytokine",0.136,3.9208,0.783,0.548,"immune response"),
c("GO:0071447","cellular response to hydroperoxide",0.001,1.3809,0.796,0.488,"immune response"),
c("GO:0001779","natural killer cell differentiation",0.004,2.0969,0.676,0.644,"immune response"),
c("GO:0033194","response to hydroperoxide",0.003,1.6402,0.820,0.376,"immune response"),
c("GO:0043353","enucleate erythrocyte differentiation",0.002,1.3507,0.635,0.609,"immune response"),
c("GO:0035691","macrophage migration inhibitory factor signaling pathway",0.000,1.4486,0.707,0.592,"immune response"),
c("GO:0006968","cellular defense response",0.002,1.7033,0.852,0.366,"immune response"),
c("GO:0042770","signal transduction in response to DNA damage",0.023,2.0531,0.688,0.434,"immune response"),
c("GO:0006952","defense response",0.568,7.2823,0.808,0.340,"immune response"),
c("GO:0006950","response to stress",4.575,4.8297,0.799,0.501,"immune response"),
c("GO:0080134","regulation of response to stress",0.337,2.1518,0.629,0.583,"immune response"),
c("GO:0044344","cellular response to fibroblast growth factor stimulus",0.031,1.4698,0.770,0.636,"immune response"),
c("GO:0031670","cellular response to nutrient",0.018,1.7055,0.758,0.587,"immune response"),
c("GO:0039530","MDA-5 signaling pathway",0.002,1.8125,0.465,0.690,"immune response"),
c("GO:0015747","urate transport",0.001,1.6968,0.963,0.000,"urate transport"),
c("GO:0043252","sodium-independent organic anion transport",0.002,1.5229,0.963,0.392,"urate transport"),
c("GO:0015881","creatine transport",0.000,1.5331,0.933,0.368,"urate transport"),
c("GO:0051674","localization of cell",0.633,1.3655,0.965,0.167,"urate transport"),
c("GO:0040011","locomotion",0.997,1.3706,0.994,0.000,"locomotion"),
c("GO:0048518","positive regulation of biological process",1.744,3.8539,0.760,0.000,"positive regulation of biological process"),
c("GO:0048522","positive regulation of cellular process",1.585,2.6459,0.655,0.323,"positive regulation of biological process"),
c("GO:0048519","negative regulation of biological process",1.984,1.6253,0.757,0.332,"positive regulation of biological process"),
c("GO:0019219","regulation of nucleobase-containing compound metabolic process",10.258,2.4498,0.644,0.643,"positive regulation of biological process"),
c("GO:0044770","cell cycle phase transition",0.207,1.3507,0.813,0.626,"positive regulation of biological process"),
c("GO:0051782","negative regulation of cell division",0.013,1.6925,0.707,0.458,"positive regulation of biological process"),
c("GO:0019222","regulation of metabolic process",11.942,2.1831,0.706,0.669,"positive regulation of biological process"),
c("GO:0050789","regulation of biological process",19.373,3.3799,0.705,0.453,"positive regulation of biological process"),
c("GO:0051336","regulation of hydrolase activity",0.855,1.3507,0.779,0.546,"positive regulation of biological process"),
c("GO:0051354","negative regulation of oxidoreductase activity",0.004,1.4559,0.743,0.417,"positive regulation of biological process"),
c("GO:2000173","negative regulation of branching morphogenesis of a nerve",0.000,1.4486,0.730,0.337,"positive regulation of biological process"),
c("GO:1900028","negative regulation of ruffle assembly",0.001,1.5376,0.689,0.542,"positive regulation of biological process"),
c("GO:0051900","regulation of mitochondrial depolarization",0.003,1.8125,0.818,0.185,"positive regulation of biological process"),
c("GO:0007162","negative regulation of cell adhesion",0.044,1.4802,0.687,0.493,"positive regulation of biological process"),
c("GO:1902750","negative regulation of cell cycle G2/M phase transition",0.027,3.3799,0.638,0.216,"positive regulation of biological process"),
c("GO:0060051","negative regulation of protein glycosylation",0.001,1.3809,0.719,0.392,"positive regulation of biological process"),
c("GO:0043902","positive regulation of multi-organism process",0.036,1.7496,0.739,0.659,"positive regulation of biological process"),
c("GO:0043900","regulation of multi-organism process",0.079,1.8894,0.803,0.237,"positive regulation of biological process"),
c("GO:0033625","positive regulation of integrin activation",0.001,2.2518,0.724,0.507,"positive regulation of biological process"),
c("GO:0044406","adhesion of symbiont to host",0.008,1.3979,0.952,0.603,"positive regulation of biological process"),
c("GO:0035306","positive regulation of dephosphorylation",0.014,1.6421,0.731,0.610,"positive regulation of biological process"),
c("GO:0040012","regulation of locomotion",0.183,1.4012,0.751,0.256,"positive regulation of biological process"),
c("GO:0045716","positive regulation of low-density lipoprotein particle receptor biosynthetic process",0.000,1.6003,0.772,0.459,"positive regulation of biological process"),
c("GO:0042752","regulation of circadian rhythm",0.028,1.3840,0.823,0.216,"positive regulation of biological process"),
c("GO:1901888","regulation of cell junction assembly",0.014,1.7959,0.712,0.538,"positive regulation of biological process"),
c("GO:0050896","response to stimulus",12.210,6.5452,0.995,0.000,"response to stimulus"),
c("GO:0051704","multi-organism process",0.751,3.3188,0.994,0.000,"multi-organism process"),
c("GO:0065007","biological regulation",20.498,3.9208,0.995,0.000,"biological regulation"),
c("GO:0033622","integrin activation",0.004,1.9393,0.905,0.020,"integrin activation"),
c("GO:0032392","DNA geometric change",0.596,1.7825,0.890,0.319,"integrin activation"),
c("GO:0070286","axonemal dynein complex assembly",0.011,1.5513,0.825,0.636,"integrin activation"),
c("GO:0071709","membrane assembly",0.095,1.3507,0.830,0.424,"integrin activation"),
c("GO:0070831","basement membrane assembly",0.004,1.3706,0.847,0.357,"integrin activation"),
c("GO:0007015","actin filament organization",0.240,1.5884,0.816,0.575,"integrin activation"),
c("GO:0016071","mRNA metabolic process",0.798,1.6021,0.911,0.036,"mRNA metabolism"),
c("GO:0072367","regulation of lipid transport by regulation of transcription from RNA polymerase II promoter",0.001,1.5186,0.758,0.536,"mRNA metabolism"),
c("GO:0072363","regulation of glycolytic process by positive regulation of transcription from RNA polymerase II promoter",0.001,1.3401,0.724,0.514,"mRNA metabolism"),
c("GO:0006366","transcription from RNA polymerase II promoter",1.430,1.3706,0.882,0.465,"mRNA metabolism"),
c("GO:0006357","regulation of transcription from RNA polymerase II promoter",1.273,1.5622,0.696,0.336,"mRNA metabolism"),
c("GO:0016070","RNA metabolic process",15.951,1.3449,0.885,0.577,"mRNA metabolism"),
c("GO:0044272","sulfur compound biosynthetic process",1.235,1.3507,0.927,0.657,"mRNA metabolism"),
c("GO:0006139","nucleobase-containing compound metabolic process",26.547,1.5817,0.897,0.597,"mRNA metabolism"),
c("GO:0006750","glutathione biosynthetic process",0.051,1.5935,0.930,0.114,"mRNA metabolism"),
c("GO:0090304","nucleic acid metabolic process",21.449,2.2373,0.887,0.306,"mRNA metabolism"),
c("GO:0008380","RNA splicing",0.413,1.3872,0.911,0.624,"mRNA metabolism"),
c("GO:0006397","mRNA processing",0.561,1.6253,0.909,0.308,"mRNA metabolism"),
c("GO:0019184","nonribosomal peptide biosynthetic process",0.076,1.3625,0.931,0.364,"mRNA metabolism"),
c("GO:0032607","interferon-alpha production",0.004,2.7352,0.846,0.050,"interferon-alpha production"),
c("GO:0051239","regulation of multicellular organismal process",0.628,1.3799,0.701,0.619,"interferon-alpha production"),
c("GO:0001816","cytokine production",0.120,1.5768,0.855,0.468,"interferon-alpha production"),
c("GO:0021543","pallium development",0.029,1.4522,0.848,0.528,"interferon-alpha production"),
c("GO:0070661","leukocyte proliferation",0.051,2.1129,0.922,0.059,"leukocyte proliferation"),
c("GO:0008283","cell proliferation",0.394,2.4112,0.940,0.069,"cell proliferation"),
c("GO:1901360","organic cyclic compound metabolic process",30.324,1.9626,0.970,0.074,"organic cyclic compound metabolism"),
c("GO:0018917","fluorene metabolic process",0.000,1.4547,0.962,0.085,"fluorene metabolism"),
c("GO:0046483","heterocycle metabolic process",29.664,1.5986,0.933,0.098,"heterocycle metabolism"),
c("GO:0006725","cellular aromatic compound metabolic process",29.628,1.5935,0.933,0.245,"heterocycle metabolism"));

stuff_down <- data.frame(revigo.data);
names(stuff_down) <- revigo.names;

stuff_down$abslog10pvalue <- as.numeric(as.character(stuff_down$abslog10pvalue));
stuff_down$freqInDbPercent <- as.numeric(as.character(stuff_down$freqInDbPercent));
stuff_down$uniqueness <- as.numeric(as.character(stuff_down$uniqueness));
stuff_down$dispensability <- as.numeric(as.character(stuff_down$dispensability));
```

### Treeplot for REVIGO data- underexpressed genes
```{r}
pdf(file="~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/gProfiler/MatLiv_MDvAlc_Downgenes_Revigo_treemap.pdf", width=16, height=9)
treemap_down <- treemap(stuff_down,
	index = c("representative","description"),
	vSize = "abslog10pvalue",
	type = "categorical",
	vColor = "representative",
	title = "REVIGO Gene Ontology treemap",
	inflate.labels = FALSE, #set this to TRUE for space-filling group labels - good for posters
	lowerbound.cex.labels = 0, #draw as many labels as possible, but some squares may not get a label)
	bg.labels = "#CCCCCCAA", #define background color of group labels, "#CCCCCC00" is fully transparent, "#CCCCCCAA" is semi-transparent grey, NA is opaque
	position.legend = "none")
dev.off()
```

# Session Information
```{r, echo = FALSE}
session_info <- devtools::session_info()
print(session_info)
```