---
title: "Maternal Mouse Liver- MD vs Alcohol - Overexpression of Pathways Analysis"
author: "Kaylee Helfrich"
date: "7/18/19"
output:
  html_document: default
  pdf_document: default
data: DEG_Results_MDvAlc_MatLiv_LFCShrink_VSTParam_RemovedNA_ExcelProof.txt
---
Built with R `r getRversion()`

```{r RMarkdown setup, include = FALSE, message = FALSE}
require(knitr)
knitr::opts_chunk$set(echo = TRUE, tidy.opts=list(width.cutoff=60), tidy=TRUE)
knitr::opts_knit$set(root.dir = "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis")
options(tinytex.verbose = TRUE)
```

# Overexpression Analysis
This document details my code and output for the overexpression analysis on maternal liver transcriptome data. Litter #'s: A1,A2,A3,A4,A5,A6,A7,A8,B1,B2,B4,B5,B6,B7,B8,B9 (where A is MD and B is Alc). 3g/kg alcohol as oral gavage (100% MD, split dose) from E8.5-17.5. Generated by KH, CK, NS Apr-June 2018. This is a direct quote to explain overrepresentation analysis: "Over-representation analysis can be summarized as follows: First, form a list of candidate genes using the marginal approach. Then, for each gene set, we create a two-by-two table comparing the number of candidate genes that are members of the category to those that are not members. The significance of over-representation can be assessed, for example, using the hypergeometric distribution or its binomial approximation. A limitation of the over-representation approach is that it ignores all the genes that did not make the list of candidate genes. Therefore, the results will be highly dependent on the cutoff used in constructing this list."

# GO Terms and Explanation
This explanation is pulled word-for-word from [this workflow](https://hbctraining.github.io/DGE_workshop/lessons/09_functional_analysis.html).
"To describe the roles of genes and gene products, GO terms are organized into three independent controlled vocabularies (ontologies) in a species-independent manner:
*Biological process: refers to the biological role involving the gene or gene product, and could include “transcription”, “signal transduction”, and “apoptosis”. A biological process generally involves a chemical or physical change of the starting material or input.
*Molecular function: represents the biochemical activity of the gene product, such activities could include “ligand”, “GTPase”, and “transporter”.
*Cellular component: refers to the location in the cell of the gene product. Cellular components could include “nucleus”, “lysosome”, and “plasma membrane”.
Each GO term has a term name (e.g. DNA repair) and a unique term accession number (GO:0005125), and a single gene product can be associated with many GO terms, since a single gene product “may function in several processes, contain domains that carry out diverse molecular functions, and participate in multiple alternative interactions with other proteins, organelles or locations in the cell”[3]." Genes can be assigned to multiple different GO terms, especially so-called hub genes, like mTOR.  All gene annotations are determined by evidence, and are updated daily. GO terms are independent of species, which can be a limitations. See [here](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003343) for more detailed information about uses and limitations of GO terms and annotations. See [here](https://yulab-smu.github.io/clusterProfiler-book/chapter5.html#supported-organisms) for explanation of how ClusterProfiler conducts the overexpression analysis (it is based on the DOSE package). Since GO is organized as a parent/child structure, outputs can be redundant by including both parents and children processes. You can simplify the outputs (see the previous link), and this is done later in this analysis. I use the package [ClusterProfiler](https://yulab-smu.github.io/clusterProfiler-book/chapter1.html) for this analysis (the term is linked for more information).

```{r include = FALSE}
rm(list=ls())
setwd("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis")
library("ggplot2")
library("BiocParallel")
library("clusterProfiler")
library("org.Mm.eg.db")
library("DOSE")
library("pathview")
library("AnnotationHub")
library("ensembldb")
library("tidyverse")
library("annotables")
library("enrichplot")
library("compEpiTools")
library("plyr")
```

## Match gene symbols with IDs
To perform subsequent steps, we need to know the actual Ensembl IDs to match the genes to the correct pathways. This step adds the correct IDs as well as further information about the genes (such as ensgene, entrez, chromosome #, start, end, strand, biotype, and description). It also creates the correct formatting for the table.
```{r Load file and add Ensembl and Entrez IDs, message = FALSE, warning = FALSE}
file <- read.table("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/DEG_Results_MDvAlc_MatLiv_LFCShrink_VSTParam_RemovedNA_ExcelProof.txt", header = TRUE, sep = "")
file <- file[,-1] #remove first column which has meaningless numbers
idx <- grcm38$symbol %in% file$GeneID #Return the IDs for the gene symbols in the DE results
ids <- grcm38[idx, ] #new dataframe with the IDs from the DE results that pulls all info from grcm38
#remove any duplicated symbols since gene names can map to more than one Ensembl ID
non_duplicates <- which(duplicated(ids$symbol) == FALSE) 
ids <- ids[non_duplicates, ] #add only the nonduplicated symbols to the file
#add the gene IDs and other information to the DEG results
file_ids <- inner_join(file, ids, by=c("GeneID"="symbol")) 
file_ids <- file_ids[,c(1,23:30,2:22)] #switch the order of the columns
```

## Create significant and background gene lists
```{r create significant and background gene lists}
all_genes <- as.character(file_ids$ensgene) #create background gene list of all labeled genes
sigOE <- filter(file_ids, padj < 0.1) #select out significantly expressed genes
sigOE_genes <- as.character(sigOE$ensgene) #create list of all significantly expressed genes
```

## Perform GO Enrichment Analysis
This analysis looks at all 3 ontologies, including molecular function, biological process, and cellular component.
```{r GOenrich on all sig diff genes}
GO_enrich <- enrichGO(gene = sigOE_genes, #significant genes for overrepresentation calculation
                universe = all_genes, #total genes measured for background level
                keyType = "ENSEMBL",
                OrgDb = org.Mm.eg.db, #compare to mouse database
                ont = "ALL", #provide MF, BP, and CC results in same file but corrected within the group
                pAdjustMethod = "BH", 
                qvalueCutoff = 0.1, 
                pvalueCutoff = 0.1,
                readable = TRUE)
                
## Output results from GO analysis to a table
cluster_summary <- data.frame(GO_enrich)
write.csv(cluster_summary, "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/MatLiv_MDvEtOH_GO_EnrichmentAnalysis.csv")
```
```{r cluster summary, echo = FALSE}
print(head(cluster_summary))
```

## Description of output file headers
The 1st and 3rd rows (no title, and "ID") contain the GO ID for the GO term. The second row ("ONTOLOGY") says which of the 3 ontologies that row describes (BP, CC, MF). The fourth row ("description") is the name/description of the GO term that the significant genes fit. The fifth row ("GeneRatio") contains the number of significantly differentially expressed genes that you provided that were annotated to that GO term over the total number of significantly differentially expressed genes that were provided AND that had GO annotations. The sixth row ("BgRatio") contains the total number of genes with that GO annotation out of the total number of background genes that you provided (significant and non-significant). The seventh row ("pvalue") contains the p-value, which is calculated using the hypergeometric function, which is [widely used](https://watermark.silverchair.com/bth456.pdf?token=AQECAHi208BE49Ooan9kkhW_Ercy7Dm3ZL_9Cf3qfKAc485ysgAAAlswggJXBgkqhkiG9w0BBwagggJIMIICRAIBADCCAj0GCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMkidHCPOxXCS-IraVAgEQgIICDjNRV7XUwDH2xnMtHQ6lcNN9ENBunYny0yBQAeW5QjULeo3DrU4tJHog4yQr42q9SO2O2khB_aOS2uZPT6-6g-GiRWbGL_FHtBBRLtRSCoh-PHfz0phz500DxbE8sHjtcQKsGfs4lY4k4IAQ1NeJvKJT_Tszx8Zxh-Wo8c2e_X1uVm1bRQB_0BNRKNDzeWAVfxoCJVty7fZfYNNgIzfJ_VgT8TaNZQZ-560ZGD4EGJnXHSoPHTkw93R4-nwN92VQQGHkwAUAVVYoZb7L9_qExCk0F1d1awHNASg2Ey65mOannd_QAyEwHUvnH7gJgapINFs6AIZiLuMF0eoGeRqR-Pbx-LwdURwKzdDwwJHxtu9GVxfXshkh1bRKySRhdHcqp8J82dg90N7TXe8vhfj47rwpj9wi-SPf8w0KWImTK6KZOCRBfmpG6DecS_x-xAf1xvkkdzSuwrWkQnZXqTZGOTwuHUP_cZsIQEuB8Pjl0KoK956CBXBnj-E1-crH52JZ5iNbbWIxU64JrtCFmw4KnjPIQQ-JZwZ7woddyaUPiniP6z-oSTNl0N0Mvq5lW49njAsK_9BTgGLpLrJryEhXr3pE-P6_WsiGvCQli8xsggSJKg0aH_T-Kl5wv5J6DaQpO7eJv9E4LM3M3ZfUc4QYhfpcKJrC87tjxVPJjLKvn8hB4Qc-d17WKgQ3ZpXpSp4). The eighth row ("p.adjust") is the p-value adjusted using Benjamini-Hochberg, as specified in the code. The ninth row ("qvalue") estimates the false-discovery rate. The tenth row ("geneID") contains all the significantly differentially expressed genes that had that particular GO term. The eleventh row ("Count") contains the total number of significantly differentially expressed genes that were annotated with that GO term.

## Create GO pathway enrichment graph
```{r create graph of all sig diff genes}
#switch the ratios in the table to decimals for easier legibility
cluster_summary$GeneRatio_Dec <-sapply(cluster_summary$GeneRatio, function(x) eval(parse(text=x)))

#Create graph of all sig expr genes and all ontologies
All <- ggplot(cluster_summary, aes(x = GeneRatio_Dec, y = fct_reorder(Description, GeneRatio_Dec))) + 
               geom_point(aes(size = Count, color = p.adjust)) +
               theme_bw(base_size = 14) +
        scale_colour_gradient(limits=c(0, 0.10), low="red") +
        ylab(NULL) +
        ggtitle(label = "GO Pathway Enrichment", 
        subtitle = "BP and CC")
```

```{r, echo = FALSE}
print(All)
```

## Interpreting the Graph
The labels on the left side of the graph indicate the [Gene Ontology process](http://amigo.geneontology.org/amigo/dd_browse) (either Biological Process (BP), Molecular Function (MF), or Cellular Component (CC)). These describe the annotation as is currently understood by previous experimental work. When a term appears, it is because there were a greater number of genes than expected that mapped to that particular ontology. The input is the list of genes that were significantly differentially expressed (I chose padj < 0.1), and the program analyzes whether more than would be expected are involved in biological processes, cellular components, or molecular functions. In this graph, the x-axis shows the gene ratio as a decimal, where 0.02 translates to 2% of the total genes that were assigned to that GO term were differentially expressed in our dataset. The size of the dots is indicated by the total number of genes that were altered, regardless of how many total genes have been annotated to that term. The color of the dots is set by the p adjusted value, where darker red indicates more significant, and more blue indicates less significant.

## Subsetting out gene ontologies
```{r subset and graph all genes based on ontology}
#Subset data based on ontology
BP <- subset(cluster_summary, ONTOLOGY == "BP")
CC <- subset(cluster_summary, ONTOLOGY == "CC")

#Create graphs
BP_graph <- ggplot(BP, aes(x = GeneRatio_Dec, y = fct_reorder(Description, GeneRatio_Dec))) + 
               geom_point(aes(size = Count, color = p.adjust)) +
               theme_bw(base_size = 14) +
        scale_colour_gradient(limits=c(0, 0.10), low="red") +
        ylab(NULL) +
        ggtitle(label = "GO Total Pathway Enrichment", 
        subtitle = "Biological Process")
```
```{r, echo = FALSE}
print(BP_graph)
```

```{r graph all genes based on ontology}
CC_graph <- ggplot(CC, aes(x = GeneRatio_Dec, y = fct_reorder(Description, GeneRatio_Dec))) + 
               geom_point(aes(size = Count, color = p.adjust)) +
               theme_bw(base_size = 14) +
        scale_colour_gradient(limits=c(0, 0.10), low="red") +
        ylab(NULL) +
        ggtitle(label = "GO Total Pathway Enrichment",
                subtitle = "Cellular Component")
```
```{r, echo = FALSE}
print(CC_graph)
```

```{r viewing all data as proportion of GO terms}
#Extract out the top of the fraction in the column GeneRatio to get diff exp genes in GO term
top <- strsplit(cluster_summary$GeneRatio, "/")
top_mat  <- matrix(unlist(top), ncol=2, byrow=TRUE)
top_vector <- top_mat[,1]
cluster_summary$top <- top_vector
#Extract out the top of the fraction in the column BgRatio to get total # of genes in GO term
bottom <- strsplit(cluster_summary$BgRatio, "/")
bottom_mat <- matrix(unlist(bottom), ncol=2, byrow=TRUE)
bottom_vector <- bottom_mat[,1]
cluster_summary$bottom <- bottom_vector
#Create fraction to see how many genes of total annotated GO genes are differentially expressed
cluster_summary$FracGO <- as.numeric(cluster_summary$top) / as.numeric(cluster_summary$bottom)
#Subset data based on ontology
BP_frac <- subset(cluster_summary, ONTOLOGY == "BP")
CC_frac <- subset(cluster_summary, ONTOLOGY == "CC")
#Create graphs
BP_frac_graph <- ggplot(BP_frac, aes(x = FracGO, y = fct_reorder(Description, FracGO))) + 
               geom_point(aes(size = Count, color = p.adjust)) +
               theme_bw(base_size = 14) +
        scale_colour_gradient(limits=c(0, 0.10), low="red") +
        ylab(NULL) +
        ggtitle(label = "Fraction of Diff Exp GO Annotated Genes", 
        subtitle = "Biological Process") +
  xlab("Fraction of diff exp GO annotated genes")
```
```{r, echo = FALSE}
print(BP_frac_graph)
```
```{r viewing all CC data as proportion of GO terms}
CC_frac_graph <- ggplot(CC_frac, aes(x = FracGO, y = fct_reorder(Description, FracGO))) + 
               geom_point(aes(size = Count, color = p.adjust)) +
               theme_bw(base_size = 14) +
        scale_colour_gradient(limits=c(0, 0.10), low="red") +
        ylab(NULL) +
        ggtitle(label = "Fraction of Diff Exp GO Annotated Genes",
                subtitle = "Cellular Component") +
  xlab("Fraction of diff exp GO annotated genes")
```
```{r, echo = FALSE}
print(CC_frac_graph)
```

## Graph Explanation
Although this graph, as before, shows the GO terms along the left side, the gene counts as the size, and the p-value in the color scale, this graph is different in exactly what data it shows. Again, only the significantly affected GO terms are shown (as indicated by p<0.1). The x-axis shows what fraction of the total genes that are annotated to a particular GO term (ex. gluconeogenesis) are differentially expressed. In other words, there is a certain number of genes in the entire dataset that are annotated as belonging to a certain GO term, and this number is the denominator, while the number of genes that were significantly different between alcohol and MD and are annotated to that same term serves as the numerator.  

# Subsetting the original data for up- and down- regulated gene graphs
The overall data can hide a lot of the information contained in the dataset. Separating out the data set into up- and down-regulated genes reveals these differences and makes the conclusions easier to understand and interpret.
```{r subset all genes by up- and down-regulated genes}
#subset  genes  by significant p-value and  by a log fold change greater than 0 (upregulated genes)
genes.up <- filter(file_ids, padj < 0.1 & log2FoldChange > 0) 
sigOE_genes.up <- as.character(genes.up$ensgene) #create a character vector of signficantly up genes
genes.down <- filter(file_ids, padj < 0.1 & log2FoldChange < 0)
sigOE_genes.down <- as.character(genes.down$ensgene)
```

## Creating GO dataframe and graph for the up-regulated genes
```{r run GOenrich on upregulated genes}
GO_enrich_up <- enrichGO(gene = sigOE_genes.up, 
                universe = all_genes,
                keyType = "ENSEMBL",
                OrgDb = org.Mm.eg.db, 
                ont = "ALL", 
                pAdjustMethod = "BH", 
                qvalueCutoff = 0.1, 
                pvalueCutoff = 0.1,
                readable = TRUE)
                
## Output results from GO analysis to a table
cluster_summary_up <- data.frame(GO_enrich_up)
write.csv(cluster_summary_up, "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/MatLiv_MDvEtOH_GO_EnrichmentAnalysis_GenesUp.csv")
```
```{r cluster summary of up-regulated genes, echo = FALSE}
print(head(cluster_summary_up))
```
```{r graph all upregulated genes}
#switch the ratios in the table to decimals
cluster_summary_up$GeneRatio_Dec <-sapply(cluster_summary_up$GeneRatio, function(x) eval(parse(text=x)))

#Create graph
All_up <- ggplot(cluster_summary_up, aes(x = GeneRatio_Dec, y = fct_reorder(Description, GeneRatio_Dec))) + 
               geom_point(aes(size = Count, color = p.adjust)) +
               theme_bw(base_size = 14) +
        scale_colour_gradient(limits=c(0, 0.10), low="red") +
        ylab(NULL) +
        ggtitle(label = "GO Upregulated Pathway Enrichment")
```
```{r, include = FALSE}
png("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/GO_CC&BP&MF_MatLiv_MDvAlc_GenesUpregulated.png", units="in", width=12, height=10, res=600) 
print(All_up)
dev.off()
```
```{r, echo = FALSE}
print(All_up)
```
```{r subset and graph upregulated genes based on BP ontology}
#Subset data based on ontology for up-regulated genes
BP_up <- subset(cluster_summary_up, ONTOLOGY == "BP")
CC_up <- subset(cluster_summary_up, ONTOLOGY == "CC")
MF_up <- subset(cluster_summary_up, ONTOLOGY == "MF")

#Create graphs
BP_graph_up <- ggplot(BP_up, aes(x = GeneRatio_Dec, y = fct_reorder(Description, GeneRatio_Dec))) + 
               geom_point(aes(size = Count, color = p.adjust)) +
               theme_bw(base_size = 14) +
        scale_colour_gradient(limits=c(0, 0.10), low="red") +
        ylab(NULL) +
  ggtitle(label = "GO Upregulated Pathway Enrichment",
              subtitle = "Biological Process") 
```
```{r, include = FALSE}
png("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/GO_BP_MatLiv_MDvAlc_GenesUpregulated.png", units="in", width=12, height=10, res=600) 
print(BP_graph_up)
dev.off()
```
```{r, echo = FALSE}
print(BP_graph_up)
```
```{r graph upregulated genes based on CC ontology}
CC_graph_up <- ggplot(CC_up, aes(x = GeneRatio_Dec, y = fct_reorder(Description, GeneRatio_Dec))) + 
               geom_point(aes(size = Count, color = p.adjust)) +
               theme_bw(base_size = 14) +
        scale_colour_gradient(limits=c(0, 0.10), low="red") +
        ylab(NULL) +
  ggtitle(label = "GO Upregulated Pathway Enrichment",
              subtitle = "Cellular Component") 
```
```{r, include = FALSE}
png("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/GO_CC_MatLiv_MDvAlc_GenesUpregulated.png", units="in", width=9, height=5, res=600) 
print(CC_graph_up)
dev.off()
```
``` {r, echo = FALSE}
print(CC_graph_up)
```
```{r graph upregulated genes based on MF ontology}
MF_graph_up <- ggplot(MF_up, aes(x = GeneRatio_Dec, y = fct_reorder(Description, GeneRatio_Dec))) + 
               geom_point(aes(size = Count, color = p.adjust)) +
               theme_bw(base_size = 14) +
        scale_colour_gradient(limits=c(0, 0.10), low="red") +
        ylab(NULL) +
  ggtitle(label = "GO Upregulated Pathway Enrichment",
              subtitle = "Molecular Function") 
```
```{r, include = FALSE}
png("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/GO_MF_MatLiv_MDvAlc_GenesUpregulated.png", units="in", width=9, height=5, res=600) 
print(MF_graph_up)
dev.off()
```
``` {r, echo = FALSE}
print(MF_graph_up)
```

```{r viewing data (upregulated genes) as proportion of GO terms}
#Extract out the top of the fraction in the column GeneRatio to get diff exp genes in GO term
top_up <- strsplit(cluster_summary_up$GeneRatio, "/")
top_mat_up  <- matrix(unlist(top_up), ncol=2, byrow=TRUE)
top_vector_up <- top_mat_up[,1]
cluster_summary_up$top <- top_vector_up
#Extract out the top of the fraction in the column BgRatio to get total # of genes in GO term
bottom_up <- strsplit(cluster_summary_up$BgRatio, "/")
bottom_mat_up <- matrix(unlist(bottom_up), ncol=2, byrow=TRUE)
bottom_vector_up <- bottom_mat_up[,1]
cluster_summary_up$bottom <- bottom_vector_up
#Create fraction to see how many genes of total annotated GO genes are differentially expressed
cluster_summary_up$FracGO <- as.numeric(cluster_summary_up$top) / as.numeric(cluster_summary_up$bottom)
#Subset data based on ontology
BP_frac_up <- subset(cluster_summary_up, ONTOLOGY == "BP")
CC_frac_up <- subset(cluster_summary_up, ONTOLOGY == "CC")
MF_frac_up <- subset(cluster_summary_up, ONTOLOGY == "MF")
#Create graphs
BP_frac_graph_up <- ggplot(BP_frac_up, aes(x = FracGO, y = fct_reorder(Description, FracGO))) + 
               geom_point(aes(size = Count, color = p.adjust)) +
               theme_bw(base_size = 12) + 
  theme(axis.title.y = element_text(size=8)) +
        scale_colour_gradient(limits=c(0, 0.10), low="red") +
        ylab(NULL) +
        ggtitle(label = "Fraction of Diff Expr GO Annotation", 
        subtitle = "Upregulated Genes-BP") +
  xlab("Fraction of diff exp GO annotated genes")
```
```{r, include = FALSE}
png("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/GO_BP_FracDiffExpGOGenes_MatLiv_GenesUpregulated.png", units="in", width=9, height=10, res=600) 
print(BP_frac_graph_up)
dev.off()
```
```{r, echo = FALSE}
print(BP_frac_graph_up)
```
```{r}
CC_frac_graph_up <- ggplot(CC_frac_up, aes(x = FracGO, y = fct_reorder(Description, FracGO))) + 
               geom_point(aes(size = Count, color = p.adjust)) +
               theme_bw(base_size = 12) +
        scale_colour_gradient(limits=c(0, 0.10), low="red") +
        ylab(NULL) +
        ggtitle(label = "Fraction of Diff Expr GO Annotation",
                subtitle = "Upregulated Genes-CC") +
  xlab("Fraction of diff exp GO annotated genes")
```
```{r, echo = FALSE}
print(CC_frac_graph_up)
```
```{r}
MF_frac_graph_up <- ggplot(MF_frac_up, aes(x = FracGO, y = fct_reorder(Description, FracGO))) + 
               geom_point(aes(size = Count, color = p.adjust)) +
               theme_bw(base_size = 12) +
        scale_colour_gradient(limits=c(0, 0.10), low="red") +
        ylab(NULL) +
        ggtitle(label = "Fraction of Diff Expr GO Annotation",
                subtitle = "Upregulated Genes-MF") +
  xlab("Fraction of diff exp GO annotated genes")
```
```{r, echo = FALSE}
print(MF_frac_graph_up)
```

## Creating a category netplot for upregulated genes to see hwo the top 10 GO terms and their associated genes interact
```{r test out a category netplot for upregulated genes}
OE_foldchanges_up <- genes.up$log2FoldChange
names(OE_foldchanges_up) <- genes.up$GeneID
cnettplot_up <- cnetplot(GO_enrich_up, 
         categorySize="pvalue", 
         showCategory = 10, #how many nodes to show
         foldChange=OE_foldchanges_up, 
         vertex.label.font=6)
```
```{r, include = FALSE}
pdf("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/MatLiv_GenesUpregulated_10NetworkPlot.pdf", width=12, height=16) 
print(cnettplot_up)
dev.off()
```
```{r, echo = FALSE}
print(cnettplot_up)
```

## Creating GO dataframe and graph for the down-regulated genes
```{r run GOenrich on downregulated genes}
GO_enrich_down <- enrichGO(gene = sigOE_genes.down, 
                universe = all_genes,
                keyType = "ENSEMBL",
                OrgDb = org.Mm.eg.db, 
                ont = "ALL", 
                pAdjustMethod = "BH", 
                qvalueCutoff = 0.1, 
                pvalueCutoff = 0.1,
                readable = TRUE)
                
## Output results from GO analysis to a table
cluster_summary_down <- data.frame(GO_enrich_down)
write.csv(cluster_summary_down, "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/MatLiv_MDvEtOH_GO_EnrichmentAnalysis_GenesDown.csv")
```
```{r cluster summary of down-regulated genes, echo = FALSE}
print(head(cluster_summary_down))
```
```{r graph all downregulated genes}
#switch the ratios in the table to decimals
cluster_summary_down$GeneRatio_Dec <-sapply(cluster_summary_down$GeneRatio, function(x) eval(parse(text=x)))

#Create graph
All_down <- ggplot(cluster_summary_down, aes(x = GeneRatio_Dec, y = fct_reorder(Description, GeneRatio_Dec))) + 
               geom_point(aes(size = Count, color = p.adjust)) +
               theme_bw(base_size = 14) +
        scale_colour_gradient(limits=c(0, 0.10), low="red") +
        ylab(NULL) +
        ggtitle("GO Downregulated Pathway Enrichment")
```
```{r, include = FALSE}
png("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/GO_CC&BP&MF_MatLiv_MDvAlc_GenesDownregulated.png", units="in", width=14, height=14, res=600) 
print(All_down)
dev.off()
```
```{r, echo = FALSE}
print(All_down)
```
```{r subset and graph based on ontology}
#Subset data based on ontology for down-regulated genes
BP_down <- subset(cluster_summary_down, ONTOLOGY == "BP")
CC_down <- subset(cluster_summary_down, ONTOLOGY == "CC")
MF_down <- subset(cluster_summary_down, ONTOLOGY == "MF")

#Create graphs
BP_graph_down <- ggplot(BP_down, aes(x = GeneRatio_Dec, y = fct_reorder(Description, GeneRatio_Dec))) + 
               geom_point(aes(size = Count, color = p.adjust)) +
               theme_bw(base_size = 14) +
        scale_colour_gradient(limits=c(0, 0.10), low="red") +
        ylab(NULL) +
  ggtitle(label = "GO Downregulated Pathway Enrichment",
              subtitle = "Biological Process") 
```
```{r, include = FALSE}
png("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/GO_BP_MatLiv_MDvAlc_GenesDownregulated.png", units="in", width=14, height=14, res=600) 
print(BP_graph_down)
dev.off()
```
```{r, echo = FALSE}
print(BP_graph_down)
```
```{r}
CC_graph_down <- ggplot(CC_down, aes(x = GeneRatio_Dec, y = fct_reorder(Description, GeneRatio_Dec))) + 
               geom_point(aes(size = Count, color = p.adjust)) +
               theme_bw(base_size = 14) +
        scale_colour_gradient(limits=c(0, 0.10), low="red") +
        ylab(NULL) +
  ggtitle(label = "GO Downregulated Pathway Enrichment",
              subtitle = "Cellular Component") 
```
```{r, include = FALSE}
png("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/GO_CC_MatLiv_MDvAlc_GenesDownregulated.png", units="in", width=9, height=5, res=600) 
print(CC_graph_down)
dev.off()
```
```{r}
print(CC_graph_down)
```
``` {r}
MF_graph_down <- ggplot(MF_down, aes(x = GeneRatio_Dec, y = fct_reorder(Description, GeneRatio_Dec))) + 
               geom_point(aes(size = Count, color = p.adjust)) +
               theme_bw(base_size = 14) +
        scale_colour_gradient(limits=c(0, 0.10), low="red") +
        ylab(NULL) +
  ggtitle(label = "GO Downregulated Pathway Enrichment",
              subtitle = "Molecular Function") 
```
```{r, include = FALSE}
png("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/GO_MF_MatLiv_MDvAlc_GenesDownregulated.png", units="in", width=12, height=10, res=600) 
print(MF_graph_down)
dev.off()
```
```{r, echo = FALSE}
print(MF_graph_down)
```

```{r viewing data (downregulated genes) as proportion of GO terms}
#Extract out the top of the fraction in the column GeneRatio to get diff exp genes in GO term
top_down <- strsplit(cluster_summary_down$GeneRatio, "/")
top_mat_down <- matrix(unlist(top_down), ncol=2, byrow=TRUE)
top_vector_down <- top_mat_down[,1]
cluster_summary_down$top <- top_vector_down
#Extract out the top of the fraction in the column BgRatio to get total # of genes in GO term
bottom_down <- strsplit(cluster_summary_down$BgRatio, "/")
bottom_mat_down <- matrix(unlist(bottom_down), ncol=2, byrow=TRUE)
bottom_vector_down <- bottom_mat_down[,1]
cluster_summary_down$bottom <- bottom_vector_down
#Create fraction to see how many genes of total annotated GO genes are differentially expressed
cluster_summary_down$FracGO <- as.numeric(cluster_summary_down$top) / as.numeric(cluster_summary_down$bottom)
#Subset data based on ontology
BP_frac_down <- subset(cluster_summary_down, ONTOLOGY == "BP")
CC_frac_down <- subset(cluster_summary_down, ONTOLOGY == "CC")
MF_frac_down <- subset(cluster_summary_down, ONTOLOGY == "MF")
#Create graphs
BP_frac_graph_down <- ggplot(BP_frac_down, aes(x = FracGO, y = fct_reorder(Description, FracGO))) + 
               geom_point(aes(size = Count, color = p.adjust)) +
               theme_bw(base_size = 14) +
        scale_colour_gradient(limits=c(0, 0.10), low="red") +
        ylab(NULL) +
        ggtitle(label = "Fraction of Diff Expr GO Annotation", 
        subtitle = "Downregulated genes- BP") +
  xlab("Fraction of diff exp GO annotated genes")
```
```{r, include = FALSE}
png("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/GO_BP_FracDiffExpGOGenes_MatLiv_GenesDownregulated.png", units="in", width=14, height=14, res=600) 
print(BP_frac_graph_down)
dev.off()
```
```{r, echo = FALSE}
print(BP_frac_graph_down)
```
```{r}
CC_frac_graph_down <- ggplot(CC_frac_down, aes(x = FracGO, y = fct_reorder(Description, FracGO))) + 
               geom_point(aes(size = Count, color = p.adjust)) +
               theme_bw(base_size = 14) +
        scale_colour_gradient(limits=c(0, 0.10), low="red") +
        ylab(NULL) +
        ggtitle(label = "Fraction of Diff Expr GO Annotation",
                subtitle = "Downregulated genes- CC") +
  xlab("Fraction of diff exp GO annotated genes")
```
```{r, include = FALSE}
png("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/GO_CC_FracDiffExpGOGenes_MatLiv_GenesDownregulated.png", units="in", width=9, height=5, res=600) 
print(CC_frac_graph_down)
dev.off()
```
```{r, echo = FALSE}
print(CC_frac_graph_down)
```
```{r}
MF_frac_graph_down <- ggplot(MF_frac_down, aes(x = FracGO, y = fct_reorder(Description, FracGO))) + 
               geom_point(aes(size = Count, color = p.adjust)) +
               theme_bw(base_size = 14) +
        scale_colour_gradient(limits=c(0, 0.10), low="red") +
        ylab(NULL) +
        ggtitle(label = "Fraction of Diff Expr GO Annotation",
                subtitle = "Downregulated genes- MF") +
  xlab("Fraction of diff exp GO annotated genes")
```
```{r, include = FALSE}
png("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/GO_MF_FracDiffExpGOGenes_MatLiv_GenesDownregulated.png", units="in", width=12, height=8, res=600) 
print(MF_frac_graph_down)
dev.off()
```
```{r, echo = FALSE}
print(MF_frac_graph_down)
```

## Creating a category netplot for downregulated genes to see how the top 10 GO terms and their associated genes interact
```{r test out a category netplot for downregulated genes}
OE_foldchanges_down <- genes.down$log2FoldChange
names(OE_foldchanges_down) <- genes.down$GeneID
cnettplot_down <- cnetplot(GO_enrich_down, 
         categorySize="pvalue", 
         showCategory = 10, #how many nodes to show
         foldChange=OE_foldchanges_down, 
         vertex.label.font=6)
```
```{r, include = FALSE}
pdf("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/MatLiv_GenesDownregulated_10NetworkPlot.pdf", width=12, height=16) 
print(cnettplot_down)
dev.off()
```
```{r, echo = FALSE}
print(cnettplot_down)
```

# Summary of Numbers Used in this Run
The original file inputted contained a total of **14,788** genes from maternal liver that were detected in the RNAseq analysis. This list excluded the genes that were not detected at all (with a 0 read count) (**5,140**), as well as the genes with low read counts (avg. read count ~less than 2) (**3,492**). I checked this list against **53,728** possible genes/variants noted in the NCBI Mus musculus 10 build (GRCm38). Matching these two lists yielded a total of **13,918** genes once duplicates had been removed (some gene names can map to more than one Ensembl ID due to the fact that some genes change ID over time). This list of **13,918** genes served as the background gene level for comparison of the differentially expressed gene list for the GO analysis (this determines whether the differentially expressed genes map to more GO terms than would be expected from background). Within this total list, there were a total of **2,110** differentially expressed genes (adjusted p-value less than 0.1), with **1,083** gene upregulated and **1,027** genes downregulated. Within the total gene list, there were **20** overexpressed GO terms (**7** BP and **13** CC). Within the upregulated gene list, there were **69** overexpressed GO terms (**56** BP,  **3** CC, and **10** MF). Within the downregulated gene list, there were **134** overexpressed GO terms (**86** BP, **22** CC, and **26** MF). 

# Rerunning analyses: Simplify the gene ontology list to decrease overlap and redundancy
This step removes redundancies. Essentially, GO terms are constructed as a hierarchy of parent and child processes. This step sees if the listed GO terms are redundant (i.e. a parent process is called because the child process is significant) and then removes the overlying parent process while leaving the more specific child process. See [here](https://rdrr.io/bioc/compEpiTools/man/simplifyGOterms.html) for more information on the function used here. 
```{r upregulated genes simplification}
#Simplify the BP ontology of upregulated genes
GOterms_BP_up <- BP_up$ID #extract out just the GO terms from the BP ontology of upregulated genes
#run the simplification pretty stringently (maxOverlap of 0.1, meaning less than 10% of genes can overlap from one to the other)
Simplified_BP_up <- simplifyGOterms(GOterms_BP_up, maxOverlap= 0.1, ontology = "BP", go2allEGs = org.Mm.egGO2ALLEGS)
Simplified_BP_up <- data.frame(Simplified_BP_up)
names(Simplified_BP_up) <- c("ID")
join_BP_up <- join(Simplified_BP_up, BP_up, by = "ID", type = "left", match = "all")

#Simplify the CC ontology of upregulated genes
GOterms_CC_up <- CC_up$ID
Simplified_CC_up <- simplifyGOterms(GOterms_CC_up, maxOverlap= 0.1, ontology = "CC", go2allEGs = org.Mm.egGO2ALLEGS)
Simplified_CC_up <- data.frame(Simplified_CC_up)
names(Simplified_CC_up) <- c("ID")
join_CC_up <- join(Simplified_CC_up, CC_up, by = "ID", type = "left", match = "all")

#Simplify the MF ontology of upregulated genes
GOterms_MF_up <- MF_up$ID
Simplified_MF_up <- simplifyGOterms(GOterms_MF_up, maxOverlap= 0.1, ontology = "MF", go2allEGs = org.Mm.egGO2ALLEGS)
Simplified_MF_up <- data.frame(Simplified_MF_up)
names(Simplified_MF_up) <- c("ID")
join_MF_up <- join(Simplified_MF_up, MF_up, by = "ID", type = "left", match = "all")

#Add together all 3 simplified lists and print out a csv of the results
Combined_simplified_up_table <- rbind(join_BP_up, join_CC_up, join_MF_up)
write.csv(Combined_simplified_up_table, "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/MatLiv_MDvEtOH_GO_EnrichmentAnalysis_GenesUp_Simplified.csv")
```
```{r summary of simplified up-regulated genes, echo = FALSE}
print(head(Combined_simplified_up_table))
```

### Results of upregulated gene simplification
This reduced the BP ontology list from **56** to **11**, the CC ontology list from **3** to **1**, and the MF ontology list from **10** to **8**. 

```{r downregulated genes simplification}
#Simplify the BP ontology of downregulated genes
GOterms_BP_down <- BP_down$ID
Simplified_BP_down <- simplifyGOterms(GOterms_BP_down, maxOverlap= 0.1, ontology = "BP", go2allEGs = org.Mm.egGO2ALLEGS)
Simplified_BP_down <- data.frame(Simplified_BP_down)
names(Simplified_BP_down) <- c("ID")
join_BP_down <- join(Simplified_BP_down, BP_down, by = "ID", type = "left", match = "all")

#Simplify the CC ontology of downregulated genes
GOterms_CC_down <- CC_down$ID
Simplified_CC_down <- simplifyGOterms(GOterms_CC_down, maxOverlap= 0.1, ontology = "CC", go2allEGs = org.Mm.egGO2ALLEGS)
Simplified_CC_down <- data.frame(Simplified_CC_down)
names(Simplified_CC_down) <- c("ID")
join_CC_down <- join(Simplified_CC_down, CC_down, by = "ID", type = "left", match = "all")

#Simplify the MF ontology of downregulated genes
GOterms_MF_down <- MF_down$ID
Simplified_MF_down <- simplifyGOterms(GOterms_MF_down, maxOverlap= 0.1, ontology = "CC", go2allEGs = org.Mm.egGO2ALLEGS)
Simplified_MF_down <- data.frame(Simplified_MF_down)
names(Simplified_MF_down) <- c("ID")
join_MF_down <- join(Simplified_MF_down, MF_down, by = "ID", type = "left", match = "all")

#Add together all 3 simplified lists and print out a csv of the results
Combined_simplified_down_table <- rbind(join_BP_down, join_CC_down, join_MF_down)
write.csv(Combined_simplified_down_table, "~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/MatLiv_MDvEtOH_GO_EnrichmentAnalysis_GenesDown_Simplified.csv")
```
```{r summary of simplified down-regulated genes, echo = FALSE}
print(head(Combined_simplified_down_table))
```

### Results of the downregulated gene simplification
This reduced the BP ontology list from **86** to **34**, the CC ontology list from **22** to **8**, and did not change the MF ontology list which remained at **26**.

### Remaking some of the upregulated genes graphs with the simplified data
These graphs are exactly the same as ones made earlier. The only difference is that they use the simplified data as generated a few lines previously. This removes some of the redundancy in the data and makes it easier to see what is happening overall with other individual pathways.
```{r}
join_BP_up$GeneRatio_Dec <-sapply(join_BP_up$GeneRatio, function(x) eval(parse(text=x)))
simple_BP_up_graph <- ggplot(join_BP_up, aes(x = GeneRatio_Dec, y = fct_reorder(Description, GeneRatio_Dec))) +
  geom_point(aes(size = Count, color = p.adjust)) + 
  theme_bw(base_size = 14) + 
  scale_colour_gradient(limits=c(0, 0.10), low="red") + 
  ylab(NULL) + 
  ggtitle(label = "GO Pathway Enrichment-BP", subtitle = ("Upregulated Genes-Simplified"))
```
```{r, echo = FALSE}
print(simple_BP_up_graph)
```
```{r}
join_CC_up$GeneRatio_Dec <-sapply(join_CC_up$GeneRatio, function(x) eval(parse(text=x)))
simple_CC_up_graph <- ggplot(join_CC_up, aes(x = GeneRatio_Dec, y = fct_reorder(Description, GeneRatio_Dec))) +
  geom_point(aes(size = Count, color = p.adjust)) + 
  theme_bw(base_size = 14) + 
  scale_colour_gradient(limits=c(0, 0.10), low="red") + 
  ylab(NULL) + 
  ggtitle(label = "GO Pathway Enrichment-CC", subtitle = ("Upregulated Genes-Simplified"))
```
```{r, echo = FALSE}
print(simple_CC_up_graph)
```
```{r}
join_MF_up$GeneRatio_Dec <-sapply(join_MF_up$GeneRatio, function(x) eval(parse(text=x)))
simple_MF_up_graph <- ggplot(join_MF_up, aes(x = GeneRatio_Dec, y = fct_reorder(Description, GeneRatio_Dec))) +
  geom_point(aes(size = Count, color = p.adjust)) + 
  theme_bw(base_size = 14) + 
  scale_colour_gradient(limits=c(0, 0.10), low="red") + 
  ylab(NULL) + 
  ggtitle(label = "GO Pathway Enrichment-MF", subtitle = ("Upregulated Genes-Simplified"))
```
```{r, echo = FALSE}
print(simple_MF_up_graph)
```

#### Creating a category netplot for the simplified upregulated genes to see how the top 10 GO terms and their associated genes interact
```{r generate category netplot for simplified upregulated genes in the BP ontology}
#regenerate the GO_enrich variable using only the BP.
GO_enrich_BP_up <- enrichGO(gene = sigOE_genes.up, 
                universe = all_genes,
                keyType = "ENSEMBL",
                OrgDb = org.Mm.eg.db, 
                ont = "BP", 
                pAdjustMethod = "BH", 
                qvalueCutoff = 0.1, 
                pvalueCutoff = 0.1,
                readable = TRUE)
cluster_summary_BPonly_up <- data.frame(GO_enrich_BP_up)
#create a new GO_enrich variable so that we don't mess up the normal one. 
GO_enrich_BP_up_simplified <- GO_enrich_BP_up
GO_simple_BP_up <- join_BP_up$ID
GO_enrich_BP_up_simplified@result <- GO_enrich_BP_up@result[c(GO_simple_BP_up),]
OE_foldchanges_simple_up <- genes.up$log2FoldChange
names(OE_foldchanges_simple_up) <- genes.up$GeneID
#create plot
cnettplot_simple_BP_up <- cnetplot(GO_enrich_BP_up_simplified, 
         categorySize="pvalue", 
         showCategory = 11, #how many nodes to show
         foldChange=OE_foldchanges_simple_up, 
         vertex.label.font=6)
```
```{r, include = FALSE}
pdf("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/Upregulated_Genes/MatLiv_BP_Simplified_GenesUpregulated_11NetworkPlot.pdf", width=12, height=16) 
print(cnettplot_simple_BP_up)
dev.off()
```
```{r, echo = FALSE}
print(cnettplot_simple_BP_up)
```
```{r generate category netplot for simplified upregulated genes in the CC ontology}
#regenerate the GO_enrich variable using only the CC. Otherwise, the analysis did not work.
GO_enrich_CC_up <- enrichGO(gene = sigOE_genes.up, 
                universe = all_genes,
                keyType = "ENSEMBL",
                OrgDb = org.Mm.eg.db, 
                ont = "CC", 
                pAdjustMethod = "BH", 
                qvalueCutoff = 0.1, 
                pvalueCutoff = 0.1,
                readable = TRUE)
cluster_summary_CConly_up <- data.frame(GO_enrich_CC_up)
#create a new GO_enrich variable so that we don't mess up the normal one. 
GO_enrich_CC_up_simplified <- GO_enrich_CC_up
GO_simple_CC_up <- join_CC_up$ID
GO_enrich_CC_up_simplified@result <- GO_enrich_CC_up@result[c(GO_simple_CC_up),]
OE_foldchanges_simple_up <- genes.up$log2FoldChange
names(OE_foldchanges_simple_up) <- genes.up$GeneID
#create plot
cnettplot_simple_CC_up <- cnetplot(GO_enrich_CC_up_simplified, 
         categorySize="pvalue", 
         showCategory = 1, #how many nodes to show
         foldChange=OE_foldchanges_simple_up, 
         vertex.label.font=6)
```
```{r, include = FALSE}
pdf("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/Upregulated_Genes/MatLiv_CC_Simplified_GenesUpregulated_1NetworkPlot.pdf", width=8, height=12) 
print(cnettplot_simple_CC_up)
dev.off()
```
```{r, echo = FALSE}
print(cnettplot_simple_CC_up)
```
```{r generate category netplot for simplified upregulated genes in the MF ontology}
#regenerate the GO_enrich variable using only the MF
GO_enrich_MF_up <- enrichGO(gene = sigOE_genes.up, 
                universe = all_genes,
                keyType = "ENSEMBL",
                OrgDb = org.Mm.eg.db, 
                ont = "MF", 
                pAdjustMethod = "BH", 
                qvalueCutoff = 0.1, 
                pvalueCutoff = 0.1,
                readable = TRUE)
cluster_summary_MFonly_up <- data.frame(GO_enrich_MF_up)
#create a new GO_enrich variable so that we don't mess up the normal one. 
GO_enrich_MF_up_simplified <- GO_enrich_MF_up
GO_simple_MF_up <- join_MF_up$ID
GO_enrich_MF_up_simplified@result <- GO_enrich_MF_up@result[c(GO_simple_MF_up),]
OE_foldchanges_simple_up <- genes.up$log2FoldChange
names(OE_foldchanges_simple_up) <- genes.up$GeneID
#create plot
cnettplot_simple_MF_up <- cnetplot(GO_enrich_MF_up_simplified, 
         categorySize="pvalue", 
         showCategory = 10, #how many nodes to show
         foldChange=OE_foldchanges_simple_up, 
         vertex.label.font=6)
```
```{r, include = FALSE}
pdf("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/Upregulated_Genes/MatLiv_MF_Simplified_GenesUpregulated_10NetworkPlot.pdf", width=12, height=16) 
print(cnettplot_simple_MF_up)
dev.off()
```
```{r, echo = FALSE}
print(cnettplot_simple_MF_up)
```


### Remaking some of the downregulated genes graphs with the simplified data
These graphs are exactly the same as ones made earlier. The only difference is that they use the simplified data as generated a few lines previously. This removes some of the redundancy in the data and makes it easier to see what is happening overall with other individual pathways.
```{r}
join_BP_down$GeneRatio_Dec <-sapply(join_BP_down$GeneRatio, function(x) eval(parse(text=x)))
simple_BP_down_graph <- ggplot(join_BP_down, aes(x = GeneRatio_Dec, y = fct_reorder(Description, GeneRatio_Dec))) +
  geom_point(aes(size = Count, color = p.adjust)) + 
  theme_bw(base_size = 14) + 
  scale_colour_gradient(limits=c(0, 0.10), low="red") + 
  ylab(NULL) + 
  ggtitle(label = "GO Pathway Enrichment-BP", subtitle = ("Downregulated Genes-Simplified"))
```
```{r, include = FALSE}
png("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/Downregulated_Genes/MatLiv_BP_Simplified_GenesDownregulated.png", units="in", width=10, height=10, res=600) 
print(simple_BP_down_graph)
dev.off()
```
```{r, echo = FALSE}
print(simple_BP_down_graph)
```
```{r}
join_CC_down$GeneRatio_Dec <-sapply(join_CC_down$GeneRatio, function(x) eval(parse(text=x)))
simple_CC_down_graph <- ggplot(join_CC_down, aes(x = GeneRatio_Dec, y = fct_reorder(Description, GeneRatio_Dec))) +
  geom_point(aes(size = Count, color = p.adjust)) + 
  theme_bw(base_size = 14) + 
  scale_colour_gradient(limits=c(0, 0.10), low="red") + 
  ylab(NULL) + 
  ggtitle(label = "GO Pathway Enrichment-CC", subtitle = ("Downregulated Genes-Simplified"))
```
```{r, echo = FALSE}
print(simple_CC_down_graph)
```

#### Creating a category netplot for the simplified downregulated genes to see how the top 10 GO terms and their associated genes interact
```{r generate category netplot for simplified downregulated genes in the BP ontology}
#regenerate the GO_enrich variable using only the BP.
GO_enrich_BP_down <- enrichGO(gene = sigOE_genes.down, 
                universe = all_genes,
                keyType = "ENSEMBL",
                OrgDb = org.Mm.eg.db, 
                ont = "BP", 
                pAdjustMethod = "BH", 
                qvalueCutoff = 0.1, 
                pvalueCutoff = 0.1,
                readable = TRUE)
cluster_summary_BPonly_down <- data.frame(GO_enrich_BP_down)
#create a new GO_enrich variable so that we don't mess up the normal one. 
GO_enrich_BP_down_simplified <- GO_enrich_BP_down
GO_simple_BP_down <- join_BP_down$ID
GO_enrich_BP_down_simplified@result <- GO_enrich_BP_down@result[c(GO_simple_BP_down),]
OE_foldchanges_simple_down <- genes.down$log2FoldChange
names(OE_foldchanges_simple_down) <- genes.down$GeneID
#create plot
cnettplot_simple_BP_down <- cnetplot(GO_enrich_BP_down_simplified, 
         categorySize="pvalue", 
         showCategory = 10, #how many nodes to show
         foldChange=OE_foldchanges_simple_down, 
         vertex.label.font=6)
```
```{r, include = FALSE}
pdf("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/Downregulated_Genes/MatLiv_BP_Simplified_GenesDownregulated_10NetworkPlot.pdf", width=12, height=16) 
print(cnettplot_simple_BP_down)
dev.off()
```
```{r, echo = FALSE}
print(cnettplot_simple_BP_down)
```
```{r generate category netplot for simplified downregulated genes in the CC ontology}
#regenerate the GO_enrich variable using only the CC. Otherwise, the analysis did not work.
GO_enrich_CC_down <- enrichGO(gene = sigOE_genes.down, 
                universe = all_genes,
                keyType = "ENSEMBL",
                OrgDb = org.Mm.eg.db, 
                ont = "CC", 
                pAdjustMethod = "BH", 
                qvalueCutoff = 0.1, 
                pvalueCutoff = 0.1,
                readable = TRUE)
cluster_summary_CConly_down <- data.frame(GO_enrich_CC_down)
#create a new GO_enrich variable so that we don't mess up the normal one. 
GO_enrich_CC_down_simplified <- GO_enrich_CC_down
GO_simple_CC_down <- join_CC_down$ID
GO_enrich_CC_down_simplified@result <- GO_enrich_CC_down@result[c(GO_simple_CC_down),]
OE_foldchanges_simple_down <- genes.down$log2FoldChange
names(OE_foldchanges_simple_down) <- genes.down$GeneID
#create plot
cnettplot_simple_CC_down <- cnetplot(GO_enrich_CC_down_simplified, 
         categorySize="pvalue", 
         showCategory = 10, #how many nodes to show
         foldChange=OE_foldchanges_simple_down, 
         vertex.label.font=6)
```
```{r, include = FALSE}
pdf("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/Downregulated_Genes/MatLiv_CC_Simplified_GenesDownregulated_10NetworkPlot.pdf", width=12, height=16) 
print(cnettplot_simple_CC_down)
dev.off()
```
```{r, echo = FALSE}
print(cnettplot_simple_CC_down)
```
```{r generate category netplot for simplified downregulated genes in the MF ontology}
#regenerate the GO_enrich variable using only the CC. Otherwise, the analysis did not work.
GO_enrich_MF_down <- enrichGO(gene = sigOE_genes.down, 
                universe = all_genes,
                keyType = "ENSEMBL",
                OrgDb = org.Mm.eg.db, 
                ont = "MF", 
                pAdjustMethod = "BH", 
                qvalueCutoff = 0.1, 
                pvalueCutoff = 0.1,
                readable = TRUE)
cluster_summary_MFonly_down <- data.frame(GO_enrich_MF_down)
#create a new GO_enrich variable so that we don't mess up the normal one. 
GO_enrich_MF_down_simplified <- GO_enrich_MF_down
GO_simple_MF_down <- join_MF_down$ID
GO_enrich_MF_down_simplified@result <- GO_enrich_MF_down@result[c(GO_simple_MF_down),]
OE_foldchanges_simple_down <- genes.down$log2FoldChange
names(OE_foldchanges_simple_down) <- genes.down$GeneID
#create plot
cnettplot_simple_MF_down <- cnetplot(GO_enrich_MF_down_simplified, 
         categorySize="pvalue", 
         showCategory = 10, #how many nodes to show
         foldChange=OE_foldchanges_simple_down, 
         vertex.label.font=6)
```
```{r, include = FALSE}
pdf("~/KayleeStuff/Smith_Lab/Data/RNA_Seq/Mat_Liv/OverexpressionAnalysis/Downregulated_Genes/MatLiv_MF_Simplified_GenesDownregulated_10NetworkPlot.pdf", width=12, height=16) 
print(cnettplot_simple_MF_down)
dev.off()
```
```{r, echo = FALSE}
print(cnettplot_simple_MF_down)
```

```{r, echo = FALSE}
session_info <- devtools::session_info()
print(session_info)
```